{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import provided tables with metadata, curate size fractions specific to Tara oceans data and generate Assembly lists"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import argparse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # import tables which list metagenomic and metatranscriptomic reads\n",
    "# metag = pd.read_csv(\"ENA_tables/PRJEB4352_metaG_wenv.txt\", sep = '\\t')\n",
    "# metat = pd.read_csv(\"ENA_tables/PRJEB6609_metaT_wenv.txt\", sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# def main():\n",
    "#     p = argparse.ArgumentParser()\n",
    "#     p.add_argument('projectfile')\n",
    "#     p.add_argument('--column',default='fastq_ftp', required=False)\n",
    "#     p.add_argument('--delimiter', default=';', required=False)\n",
    "#     args = p.parse_args() \n",
    "#     #read in file and split the ftp column\n",
    "#     df = pd.read_table(args.projectfile) \n",
    "#     df[['c1', 'c2']]=df[args.column].str.split(args.delimiter, expand=True)\n",
    "#     #test if PE read by looking for fastq2 file\n",
    "#     pe_ind = df.c2.notnull()    \n",
    "#     se_ind = df.c2.isnull()\n",
    "#     basename = args.projectfile.split('.')[0]\n",
    "#     PE_out = basename + '_PE.txt'\n",
    "#     SE_out = basename + '_SE.txt'\n",
    "#     PE_READS = df.loc[pe_ind].drop(['c1', 'c2'], axis=1)\n",
    "#     SE_READS = df.loc[se_ind].drop(['c1', 'c2'], axis=1)  \n",
    "#     PE_READS.to_csv(PE_out, sep='\\t') \n",
    "#     SE_READS.to_csv(SE_out, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import tables which list metagenomic and metatranscriptomic reads\n",
    "metag = pd.read_csv(\"ENA_tables/PRJEB4352_metaG_wenv_PE.txt\", sep = '\\t')\n",
    "metat = pd.read_csv(\"ENA_tables/PRJEB6609_metaT_wenv_PE.txt\", sep = '\\t')\n",
    "\n",
    "# Generate new columns called size frac\n",
    "metag['size_frac']= metag['Fraction lower [µm]'].map(str) + '-' + metag['Fraction upper [µm]'].map(str)\n",
    "metat['size_frac']= metat['Fraction lower [µm]'].map(str) + '-' + metat['Fraction upper [µm]'].map(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>study_accession</th>\n",
       "      <th>sample_accession</th>\n",
       "      <th>secondary_sample_accession</th>\n",
       "      <th>experiment_accession</th>\n",
       "      <th>run_accession</th>\n",
       "      <th>tax_id</th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>...</th>\n",
       "      <th>v [cm/s]</th>\n",
       "      <th>OW</th>\n",
       "      <th>MLE [1/day]</th>\n",
       "      <th>RT [days]</th>\n",
       "      <th>Latitude (of the continental shelf with...)</th>\n",
       "      <th>Longitude (of the continental shelf with...)</th>\n",
       "      <th>Age [days]</th>\n",
       "      <th>Latitude (of the geographic coordinate ...)</th>\n",
       "      <th>Longitude (of the geographic coordinate ...)</th>\n",
       "      <th>size_frac</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2656965</td>\n",
       "      <td>ERS488900</td>\n",
       "      <td>ERX1771177</td>\n",
       "      <td>ERR1700889</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>15.046586</td>\n",
       "      <td>0.403157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.250000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>16.583327</td>\n",
       "      <td>63.625415</td>\n",
       "      <td>180.0-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620205</td>\n",
       "      <td>ERS489054</td>\n",
       "      <td>ERX1771189</td>\n",
       "      <td>ERR1700890</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>4.306080</td>\n",
       "      <td>-0.412443</td>\n",
       "      <td>0.026178</td>\n",
       "      <td>19.027778</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>14.404658</td>\n",
       "      <td>69.689377</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623506</td>\n",
       "      <td>ERS493999</td>\n",
       "      <td>ERX1771187</td>\n",
       "      <td>ERR1700891</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>8.317923</td>\n",
       "      <td>-2.922481</td>\n",
       "      <td>0.044572</td>\n",
       "      <td>25.750000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>26.338683</td>\n",
       "      <td>-87.272865</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620172</td>\n",
       "      <td>ERS489013</td>\n",
       "      <td>ERX1771191</td>\n",
       "      <td>ERR1700892</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-33.418508</td>\n",
       "      <td>0.043242</td>\n",
       "      <td>0.086921</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>17.386625</td>\n",
       "      <td>66.210146</td>\n",
       "      <td>3.0-&gt;3.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2730950</td>\n",
       "      <td>ERS494049</td>\n",
       "      <td>ERX1771202</td>\n",
       "      <td>ERR1700893</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>91.787857</td>\n",
       "      <td>-4.707638</td>\n",
       "      <td>0.398799</td>\n",
       "      <td>1.083333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619554</td>\n",
       "      <td>ERS488153</td>\n",
       "      <td>ERX1771192</td>\n",
       "      <td>ERR1700894</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>4.368810</td>\n",
       "      <td>0.021447</td>\n",
       "      <td>0.190651</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>38.821749</td>\n",
       "      <td>6.902024</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619623</td>\n",
       "      <td>ERS488251</td>\n",
       "      <td>ERX1771178</td>\n",
       "      <td>ERR1700895</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.797800</td>\n",
       "      <td>41.665900</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>7</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619725</td>\n",
       "      <td>ERS488412</td>\n",
       "      <td>ERX1771195</td>\n",
       "      <td>ERR1700896</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.375367</td>\n",
       "      <td>0.051438</td>\n",
       "      <td>0.087535</td>\n",
       "      <td>1.333333</td>\n",
       "      <td>14.973150</td>\n",
       "      <td>34.451200</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32.144171</td>\n",
       "      <td>16.847326</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>8</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622039</td>\n",
       "      <td>ERS491758</td>\n",
       "      <td>ERX1771199</td>\n",
       "      <td>ERR1700897</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>1.275138</td>\n",
       "      <td>-0.006839</td>\n",
       "      <td>0.028784</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-25.231969</td>\n",
       "      <td>-112.818256</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2621474</td>\n",
       "      <td>ERS490981</td>\n",
       "      <td>ERX1771201</td>\n",
       "      <td>ERR1700898</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>1.423920</td>\n",
       "      <td>0.072135</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-65.022700</td>\n",
       "      <td>-54.418000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620072</td>\n",
       "      <td>ERS488886</td>\n",
       "      <td>ERX1771182</td>\n",
       "      <td>ERR1700899</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>15.046586</td>\n",
       "      <td>0.403157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.250000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>16.583327</td>\n",
       "      <td>63.625415</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>11</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619421</td>\n",
       "      <td>ERS487958</td>\n",
       "      <td>ERX1771180</td>\n",
       "      <td>ERR1700900</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.031746</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>-6.552800</td>\n",
       "      <td>36.563150</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>12</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2621827</td>\n",
       "      <td>ERS491478</td>\n",
       "      <td>ERX1771186</td>\n",
       "      <td>ERR1700901</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.678504</td>\n",
       "      <td>0.061918</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.250000</td>\n",
       "      <td>-72.642651</td>\n",
       "      <td>-35.181114</td>\n",
       "      <td>63.0</td>\n",
       "      <td>-34.302478</td>\n",
       "      <td>-73.496846</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>13</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2731011</td>\n",
       "      <td>ERS494136</td>\n",
       "      <td>ERX1771181</td>\n",
       "      <td>ERR1700902</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>34.457153</td>\n",
       "      <td>2.028463</td>\n",
       "      <td>0.249201</td>\n",
       "      <td>1.861111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>29.041205</td>\n",
       "      <td>-79.321465</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>14</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622132</td>\n",
       "      <td>ERS491887</td>\n",
       "      <td>ERX1771196</td>\n",
       "      <td>ERR1700903</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>2.208339</td>\n",
       "      <td>0.013354</td>\n",
       "      <td>0.045278</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-13.259274</td>\n",
       "      <td>-94.231260</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>15</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619959</td>\n",
       "      <td>ERS488754</td>\n",
       "      <td>ERX1771197</td>\n",
       "      <td>ERR1700904</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>8.325847</td>\n",
       "      <td>-0.030610</td>\n",
       "      <td>0.074837</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>23.584126</td>\n",
       "      <td>60.241719</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>16</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622039</td>\n",
       "      <td>ERS491758</td>\n",
       "      <td>ERX1771185</td>\n",
       "      <td>ERR1700905</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>1.275138</td>\n",
       "      <td>-0.006839</td>\n",
       "      <td>0.028784</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-25.231969</td>\n",
       "      <td>-112.818256</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>17</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2657022</td>\n",
       "      <td>ERS488820</td>\n",
       "      <td>ERX1771190</td>\n",
       "      <td>ERR1700906</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.652039</td>\n",
       "      <td>0.005785</td>\n",
       "      <td>0.069484</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>20.258041</td>\n",
       "      <td>64.562537</td>\n",
       "      <td>180.0-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>18</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2657022</td>\n",
       "      <td>ERS488820</td>\n",
       "      <td>ERX1771188</td>\n",
       "      <td>ERR1700907</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.652039</td>\n",
       "      <td>0.005785</td>\n",
       "      <td>0.069484</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>20.258041</td>\n",
       "      <td>64.562537</td>\n",
       "      <td>180.0-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>19</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619421</td>\n",
       "      <td>ERS487958</td>\n",
       "      <td>ERX1771198</td>\n",
       "      <td>ERR1700908</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.031746</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>-6.552800</td>\n",
       "      <td>36.563150</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>20</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2657050</td>\n",
       "      <td>ERS489336</td>\n",
       "      <td>ERX1771193</td>\n",
       "      <td>ERR1700909</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>19.395807</td>\n",
       "      <td>-0.118280</td>\n",
       "      <td>0.067544</td>\n",
       "      <td>3.777778</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-14.979369</td>\n",
       "      <td>71.710551</td>\n",
       "      <td>180.0-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>21</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2732453</td>\n",
       "      <td>ERS493776</td>\n",
       "      <td>ERX1771200</td>\n",
       "      <td>ERR1700910</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>1.040314</td>\n",
       "      <td>-0.002795</td>\n",
       "      <td>0.020154</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>7.512087</td>\n",
       "      <td>-111.603363</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>22</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2657097</td>\n",
       "      <td>ERS492000</td>\n",
       "      <td>ERX1771184</td>\n",
       "      <td>ERR1700911</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>15.658745</td>\n",
       "      <td>0.043188</td>\n",
       "      <td>0.012324</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-9.444510</td>\n",
       "      <td>-86.887739</td>\n",
       "      <td>180.0-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>23</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620222</td>\n",
       "      <td>ERS489079</td>\n",
       "      <td>ERX1771194</td>\n",
       "      <td>ERR1700912</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>4.306080</td>\n",
       "      <td>-0.412443</td>\n",
       "      <td>0.026178</td>\n",
       "      <td>19.027778</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>14.404658</td>\n",
       "      <td>69.689377</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>24</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619789</td>\n",
       "      <td>ERS488516</td>\n",
       "      <td>ERX1771183</td>\n",
       "      <td>ERR1700913</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>5.586469</td>\n",
       "      <td>0.369749</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>25</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2656981</td>\n",
       "      <td>ERS488816</td>\n",
       "      <td>ERX1771179</td>\n",
       "      <td>ERR1700914</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.652039</td>\n",
       "      <td>0.005785</td>\n",
       "      <td>0.069484</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>20.258041</td>\n",
       "      <td>64.562537</td>\n",
       "      <td>20.0-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>26</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2657109</td>\n",
       "      <td>ERS492751</td>\n",
       "      <td>ERX1796605</td>\n",
       "      <td>ERR1726522</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>2.890793</td>\n",
       "      <td>-0.011503</td>\n",
       "      <td>0.053774</td>\n",
       "      <td>1.083333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-8.878592</td>\n",
       "      <td>-137.774399</td>\n",
       "      <td>5.0-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>27</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623302</td>\n",
       "      <td>ERS493677</td>\n",
       "      <td>ERX1796551</td>\n",
       "      <td>ERR1726523</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>3.470203</td>\n",
       "      <td>0.075315</td>\n",
       "      <td>0.032361</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>12.005218</td>\n",
       "      <td>-114.248528</td>\n",
       "      <td>0.8-&gt;0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>28</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2621794</td>\n",
       "      <td>ERS491436</td>\n",
       "      <td>ERX1796959</td>\n",
       "      <td>ERR1726524</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.678504</td>\n",
       "      <td>0.061918</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.250000</td>\n",
       "      <td>-72.642651</td>\n",
       "      <td>-35.181114</td>\n",
       "      <td>63.0</td>\n",
       "      <td>-34.302478</td>\n",
       "      <td>-73.496846</td>\n",
       "      <td>20.0-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>29</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2732548</td>\n",
       "      <td>ERS492060</td>\n",
       "      <td>ERX1796611</td>\n",
       "      <td>ERR1726525</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>3.096856</td>\n",
       "      <td>-0.041450</td>\n",
       "      <td>0.138825</td>\n",
       "      <td>4.166667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>4.631580</td>\n",
       "      <td>-82.454814</td>\n",
       "      <td>20.0-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>792</th>\n",
       "      <td>793</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623667</td>\n",
       "      <td>ERS494226</td>\n",
       "      <td>ERX948013</td>\n",
       "      <td>ERR868495</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>4.126204</td>\n",
       "      <td>-0.044917</td>\n",
       "      <td>0.144593</td>\n",
       "      <td>3.472222</td>\n",
       "      <td>-70.706411</td>\n",
       "      <td>39.957414</td>\n",
       "      <td>24.0</td>\n",
       "      <td>39.495355</td>\n",
       "      <td>-69.744626</td>\n",
       "      <td>0.8-3.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>793</th>\n",
       "      <td>794</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619754</td>\n",
       "      <td>ERS488460</td>\n",
       "      <td>ERX948027</td>\n",
       "      <td>ERR868496</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>16.845059</td>\n",
       "      <td>0.054261</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>794</th>\n",
       "      <td>795</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619443</td>\n",
       "      <td>ERS487982</td>\n",
       "      <td>ERX947980</td>\n",
       "      <td>ERR868497</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>24.270961</td>\n",
       "      <td>-10.417380</td>\n",
       "      <td>0.284530</td>\n",
       "      <td>13.638889</td>\n",
       "      <td>-4.201844</td>\n",
       "      <td>35.489171</td>\n",
       "      <td>42.0</td>\n",
       "      <td>35.673706</td>\n",
       "      <td>-4.145546</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>795</th>\n",
       "      <td>796</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620010</td>\n",
       "      <td>ERS488809</td>\n",
       "      <td>ERX948026</td>\n",
       "      <td>ERR868498</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.652039</td>\n",
       "      <td>0.005785</td>\n",
       "      <td>0.069484</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>20.258041</td>\n",
       "      <td>64.562537</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>796</th>\n",
       "      <td>797</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622010</td>\n",
       "      <td>ERS491719</td>\n",
       "      <td>ERX948101</td>\n",
       "      <td>ERR868499</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>1.275138</td>\n",
       "      <td>-0.006839</td>\n",
       "      <td>0.028784</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-25.231969</td>\n",
       "      <td>-112.818256</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>797</th>\n",
       "      <td>798</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622997</td>\n",
       "      <td>ERS493199</td>\n",
       "      <td>ERX948090</td>\n",
       "      <td>ERR868500</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.599451</td>\n",
       "      <td>-0.181183</td>\n",
       "      <td>0.100184</td>\n",
       "      <td>1.611111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>11.448054</td>\n",
       "      <td>-150.695800</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>798</th>\n",
       "      <td>799</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622536</td>\n",
       "      <td>ERS492426</td>\n",
       "      <td>ERX947963</td>\n",
       "      <td>ERR868501</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.682669</td>\n",
       "      <td>-0.120564</td>\n",
       "      <td>0.040113</td>\n",
       "      <td>11.083333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-23.568914</td>\n",
       "      <td>-127.310923</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>799</th>\n",
       "      <td>800</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622258</td>\n",
       "      <td>ERS492052</td>\n",
       "      <td>ERX947951</td>\n",
       "      <td>ERR868502</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>3.096856</td>\n",
       "      <td>-0.041450</td>\n",
       "      <td>0.138825</td>\n",
       "      <td>4.166667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>4.631580</td>\n",
       "      <td>-82.454814</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>800</th>\n",
       "      <td>801</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620010</td>\n",
       "      <td>ERS488809</td>\n",
       "      <td>ERX948057</td>\n",
       "      <td>ERR868503</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.652039</td>\n",
       "      <td>0.005785</td>\n",
       "      <td>0.069484</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>20.258041</td>\n",
       "      <td>64.562537</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>801</th>\n",
       "      <td>802</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619723</td>\n",
       "      <td>ERS488410</td>\n",
       "      <td>ERX948015</td>\n",
       "      <td>ERR868504</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.375367</td>\n",
       "      <td>0.051438</td>\n",
       "      <td>0.087535</td>\n",
       "      <td>1.333333</td>\n",
       "      <td>14.973150</td>\n",
       "      <td>34.451200</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32.144171</td>\n",
       "      <td>16.847326</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>802</th>\n",
       "      <td>803</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622278</td>\n",
       "      <td>ERS492090</td>\n",
       "      <td>ERX948003</td>\n",
       "      <td>ERR868505</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>3.096856</td>\n",
       "      <td>-0.041450</td>\n",
       "      <td>0.138825</td>\n",
       "      <td>4.166667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>4.631580</td>\n",
       "      <td>-82.454814</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>803</th>\n",
       "      <td>804</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622369</td>\n",
       "      <td>ERS492212</td>\n",
       "      <td>ERX948066</td>\n",
       "      <td>ERR868506</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.243588</td>\n",
       "      <td>-0.002594</td>\n",
       "      <td>0.005026</td>\n",
       "      <td>2.083333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>6.165688</td>\n",
       "      <td>-92.915797</td>\n",
       "      <td>0.8-3.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>804</th>\n",
       "      <td>805</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620354</td>\n",
       "      <td>ERS489251</td>\n",
       "      <td>ERX948006</td>\n",
       "      <td>ERR868507</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-18.737291</td>\n",
       "      <td>0.111952</td>\n",
       "      <td>0.069572</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>72.715398</td>\n",
       "      <td>3.753862</td>\n",
       "      <td>31.0</td>\n",
       "      <td>2.260611</td>\n",
       "      <td>75.796797</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>805</th>\n",
       "      <td>806</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620292</td>\n",
       "      <td>ERS489175</td>\n",
       "      <td>ERX947995</td>\n",
       "      <td>ERR868508</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.593482</td>\n",
       "      <td>0.146439</td>\n",
       "      <td>0.118141</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>73.488950</td>\n",
       "      <td>4.660350</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>806</th>\n",
       "      <td>807</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623564</td>\n",
       "      <td>ERS494087</td>\n",
       "      <td>ERX948073</td>\n",
       "      <td>ERR868509</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>91.787857</td>\n",
       "      <td>-4.707638</td>\n",
       "      <td>0.398799</td>\n",
       "      <td>1.083333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>808</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619419</td>\n",
       "      <td>ERS487956</td>\n",
       "      <td>ERX947959</td>\n",
       "      <td>ERR868510</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.031746</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>-6.552800</td>\n",
       "      <td>36.563150</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>809</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623839</td>\n",
       "      <td>ERS494501</td>\n",
       "      <td>ERX948078</td>\n",
       "      <td>ERR868511</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>4.559712</td>\n",
       "      <td>-0.032539</td>\n",
       "      <td>0.073051</td>\n",
       "      <td>1.388889</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>36.403496</td>\n",
       "      <td>-38.208687</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>809</th>\n",
       "      <td>810</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623321</td>\n",
       "      <td>ERS493712</td>\n",
       "      <td>ERX948018</td>\n",
       "      <td>ERR868512</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>3.470203</td>\n",
       "      <td>0.075315</td>\n",
       "      <td>0.032361</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>12.005218</td>\n",
       "      <td>-114.248528</td>\n",
       "      <td>0.8-3.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>811</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622661</td>\n",
       "      <td>ERS492651</td>\n",
       "      <td>ERX948010</td>\n",
       "      <td>ERR868513</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>5.403076</td>\n",
       "      <td>0.007751</td>\n",
       "      <td>0.028977</td>\n",
       "      <td>15.833333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-11.599533</td>\n",
       "      <td>-137.896881</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>812</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622679</td>\n",
       "      <td>ERS492686</td>\n",
       "      <td>ERX953564</td>\n",
       "      <td>ERR873959</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>5.403076</td>\n",
       "      <td>0.007751</td>\n",
       "      <td>0.028977</td>\n",
       "      <td>15.833333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-11.599533</td>\n",
       "      <td>-137.896881</td>\n",
       "      <td>0.8-3.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>812</th>\n",
       "      <td>813</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623246</td>\n",
       "      <td>ERS493592</td>\n",
       "      <td>ERX953568</td>\n",
       "      <td>ERR873960</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>1.721794</td>\n",
       "      <td>-1.155182</td>\n",
       "      <td>0.047316</td>\n",
       "      <td>41.666667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>33.472302</td>\n",
       "      <td>-120.652510</td>\n",
       "      <td>0.8-3.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>813</th>\n",
       "      <td>814</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622038</td>\n",
       "      <td>ERS491757</td>\n",
       "      <td>ERX953562</td>\n",
       "      <td>ERR873961</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>1.275138</td>\n",
       "      <td>-0.006839</td>\n",
       "      <td>0.028784</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-25.231969</td>\n",
       "      <td>-112.818256</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>814</th>\n",
       "      <td>815</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619981</td>\n",
       "      <td>ERS488780</td>\n",
       "      <td>ERX953565</td>\n",
       "      <td>ERR873962</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>8.555371</td>\n",
       "      <td>-0.087532</td>\n",
       "      <td>0.061870</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>61.900938</td>\n",
       "      <td>24.819492</td>\n",
       "      <td>50.0</td>\n",
       "      <td>24.428166</td>\n",
       "      <td>61.449162</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>815</th>\n",
       "      <td>816</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622511</td>\n",
       "      <td>ERS492393</td>\n",
       "      <td>ERX953572</td>\n",
       "      <td>ERR873963</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>6.423670</td>\n",
       "      <td>-0.075767</td>\n",
       "      <td>0.018781</td>\n",
       "      <td>34.305556</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-16.755721</td>\n",
       "      <td>-99.729821</td>\n",
       "      <td>0.8-3.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>816</th>\n",
       "      <td>817</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623603</td>\n",
       "      <td>ERS494131</td>\n",
       "      <td>ERX953566</td>\n",
       "      <td>ERR873964</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>34.457153</td>\n",
       "      <td>2.028463</td>\n",
       "      <td>0.249201</td>\n",
       "      <td>1.861111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>29.041205</td>\n",
       "      <td>-79.321465</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>817</th>\n",
       "      <td>818</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623579</td>\n",
       "      <td>ERS494105</td>\n",
       "      <td>ERX953567</td>\n",
       "      <td>ERR873965</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>91.787857</td>\n",
       "      <td>-4.707638</td>\n",
       "      <td>0.398799</td>\n",
       "      <td>1.083333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8-3.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>818</th>\n",
       "      <td>819</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622038</td>\n",
       "      <td>ERS491757</td>\n",
       "      <td>ERX953571</td>\n",
       "      <td>ERR873966</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>1.275138</td>\n",
       "      <td>-0.006839</td>\n",
       "      <td>0.028784</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-25.231969</td>\n",
       "      <td>-112.818256</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>819</th>\n",
       "      <td>820</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623246</td>\n",
       "      <td>ERS493592</td>\n",
       "      <td>ERX953569</td>\n",
       "      <td>ERR873967</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>1.721794</td>\n",
       "      <td>-1.155182</td>\n",
       "      <td>0.047316</td>\n",
       "      <td>41.666667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>33.472302</td>\n",
       "      <td>-120.652510</td>\n",
       "      <td>0.8-3.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>820</th>\n",
       "      <td>821</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622679</td>\n",
       "      <td>ERS492686</td>\n",
       "      <td>ERX953570</td>\n",
       "      <td>ERR873968</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>5.403076</td>\n",
       "      <td>0.007751</td>\n",
       "      <td>0.028977</td>\n",
       "      <td>15.833333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-11.599533</td>\n",
       "      <td>-137.896881</td>\n",
       "      <td>0.8-3.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>821</th>\n",
       "      <td>822</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619981</td>\n",
       "      <td>ERS488780</td>\n",
       "      <td>ERX953563</td>\n",
       "      <td>ERR873969</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>8.555371</td>\n",
       "      <td>-0.087532</td>\n",
       "      <td>0.061870</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>61.900938</td>\n",
       "      <td>24.819492</td>\n",
       "      <td>50.0</td>\n",
       "      <td>24.428166</td>\n",
       "      <td>61.449162</td>\n",
       "      <td>0.8-5.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>822 rows × 97 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0 study_accession sample_accession secondary_sample_accession  \\\n",
       "0             0       PRJEB4352     SAMEA2656965                  ERS488900   \n",
       "1             1       PRJEB4352     SAMEA2620205                  ERS489054   \n",
       "2             2       PRJEB4352     SAMEA2623506                  ERS493999   \n",
       "3             3       PRJEB4352     SAMEA2620172                  ERS489013   \n",
       "4             4       PRJEB4352     SAMEA2730950                  ERS494049   \n",
       "..          ...             ...              ...                        ...   \n",
       "817         818       PRJEB4352     SAMEA2623579                  ERS494105   \n",
       "818         819       PRJEB4352     SAMEA2622038                  ERS491757   \n",
       "819         820       PRJEB4352     SAMEA2623246                  ERS493592   \n",
       "820         821       PRJEB4352     SAMEA2622679                  ERS492686   \n",
       "821         822       PRJEB4352     SAMEA2619981                  ERS488780   \n",
       "\n",
       "    experiment_accession run_accession  tax_id    scientific_name  \\\n",
       "0             ERX1771177    ERR1700889  408172  marine metagenome   \n",
       "1             ERX1771189    ERR1700890  408172  marine metagenome   \n",
       "2             ERX1771187    ERR1700891  408172  marine metagenome   \n",
       "3             ERX1771191    ERR1700892  408172  marine metagenome   \n",
       "4             ERX1771202    ERR1700893  408172  marine metagenome   \n",
       "..                   ...           ...     ...                ...   \n",
       "817            ERX953567     ERR873965  408172  marine metagenome   \n",
       "818            ERX953571     ERR873966  408172  marine metagenome   \n",
       "819            ERX953569     ERR873967  408172  marine metagenome   \n",
       "820            ERX953570     ERR873968  408172  marine metagenome   \n",
       "821            ERX953563     ERR873969  408172  marine metagenome   \n",
       "\n",
       "        instrument_model library_layout  ...   v [cm/s]        OW MLE [1/day]  \\\n",
       "0    Illumina HiSeq 2000         PAIRED  ...  15.046586  0.403157         NaN   \n",
       "1    Illumina HiSeq 2000         PAIRED  ...   4.306080 -0.412443    0.026178   \n",
       "2    Illumina HiSeq 2000         PAIRED  ...   8.317923 -2.922481    0.044572   \n",
       "3    Illumina HiSeq 2000         PAIRED  ... -33.418508  0.043242    0.086921   \n",
       "4    Illumina HiSeq 2000         PAIRED  ...  91.787857 -4.707638    0.398799   \n",
       "..                   ...            ...  ...        ...       ...         ...   \n",
       "817  Illumina HiSeq 2000         PAIRED  ...  91.787857 -4.707638    0.398799   \n",
       "818  Illumina HiSeq 2000         PAIRED  ...   1.275138 -0.006839    0.028784   \n",
       "819  Illumina HiSeq 2000         PAIRED  ...   1.721794 -1.155182    0.047316   \n",
       "820  Illumina HiSeq 2000         PAIRED  ...   5.403076  0.007751    0.028977   \n",
       "821  Illumina HiSeq 2000         PAIRED  ...   8.555371 -0.087532    0.061870   \n",
       "\n",
       "     RT [days] Latitude (of the continental shelf with...)  \\\n",
       "0     2.250000                                         NaN   \n",
       "1    19.027778                                         NaN   \n",
       "2    25.750000                                         NaN   \n",
       "3     0.000000                                         NaN   \n",
       "4     1.083333                                         NaN   \n",
       "..         ...                                         ...   \n",
       "817   1.083333                                         NaN   \n",
       "818   0.000000                                         NaN   \n",
       "819  41.666667                                         NaN   \n",
       "820  15.833333                                         NaN   \n",
       "821   0.000000                                   61.900938   \n",
       "\n",
       "    Longitude (of the continental shelf with...) Age [days]  \\\n",
       "0                                            NaN      120.0   \n",
       "1                                            NaN      120.0   \n",
       "2                                            NaN      120.0   \n",
       "3                                            NaN      120.0   \n",
       "4                                            NaN      120.0   \n",
       "..                                           ...        ...   \n",
       "817                                          NaN      120.0   \n",
       "818                                          NaN      120.0   \n",
       "819                                          NaN      120.0   \n",
       "820                                          NaN      120.0   \n",
       "821                                    24.819492       50.0   \n",
       "\n",
       "    Latitude (of the geographic coordinate ...)  \\\n",
       "0                                     16.583327   \n",
       "1                                     14.404658   \n",
       "2                                     26.338683   \n",
       "3                                     17.386625   \n",
       "4                                           NaN   \n",
       "..                                          ...   \n",
       "817                                         NaN   \n",
       "818                                  -25.231969   \n",
       "819                                   33.472302   \n",
       "820                                  -11.599533   \n",
       "821                                   24.428166   \n",
       "\n",
       "    Longitude (of the geographic coordinate ...)      size_frac  \n",
       "0                                      63.625415  180.0-2000.00  \n",
       "1                                      69.689377      5.0-20.00  \n",
       "2                                     -87.272865      5.0-20.00  \n",
       "3                                      66.210146      3.0->3.00  \n",
       "4                                            NaN      5.0-20.00  \n",
       "..                                           ...            ...  \n",
       "817                                          NaN       0.8-3.00  \n",
       "818                                  -112.818256       0.8-5.00  \n",
       "819                                  -120.652510       0.8-3.00  \n",
       "820                                  -137.896881       0.8-3.00  \n",
       "821                                    61.449162       0.8-5.00  \n",
       "\n",
       "[822 rows x 97 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metag"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Tidy Size fraction data splitting "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>size_frac</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0.22-3.00</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.8-20.00</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.8-3.00</th>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.8-5.00</th>\n",
       "      <td>204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.8-&gt;0.80</th>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180.0-2000.00</th>\n",
       "      <td>179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20.0-180.00</th>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3.0-&gt;3.00</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3.0-&gt;5.00</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5.0-20.00</th>\n",
       "      <td>136</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               study_accession\n",
       "size_frac                     \n",
       "0.22-3.00                    1\n",
       "0.8-20.00                    7\n",
       "0.8-3.00                    34\n",
       "0.8-5.00                   204\n",
       "0.8->0.80                   79\n",
       "180.0-2000.00              179\n",
       "20.0-180.00                167\n",
       "3.0->3.00                   14\n",
       "3.0->5.00                    1\n",
       "5.0-20.00                  136"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Original data -- group by size fractions in 4 distinct groups\n",
    "pd.DataFrame(metag.groupby('size_frac').count()['study_accession'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>size_frac</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0.8-5.00</th>\n",
       "      <td>339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180-2000.00</th>\n",
       "      <td>179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20-180.00</th>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5-20.00</th>\n",
       "      <td>136</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             study_accession\n",
       "size_frac                   \n",
       "0.8-5.00                 339\n",
       "180-2000.00              179\n",
       "20-180.00                167\n",
       "5-20.00                  136"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#drop 0.22 sample\n",
    "metag = metag[metag.size_frac != '0.22-3.00']\n",
    "metag = metag.replace('0.8->0.80', '0.8-5.00')\n",
    "metag = metag.replace('0.8-20.00', '0.8-5.00')\n",
    "metag = metag.replace('0.8-3.00', '0.8-5.00')\n",
    "metag = metag.replace('3.0->3.00', '0.8-5.00')\n",
    "metag = metag.replace('3.0->5.00', '0.8-5.00')\n",
    "metag = metag.replace('180.0-2000.00', '180-2000.00')\n",
    "metag = metag.replace('20.0-180.00', '20-180.00')\n",
    "metag = metag.replace('5.0-20.00', '5-20.00')\n",
    "\n",
    "#Check final groupings\n",
    "pd.DataFrame(metag.groupby('size_frac').count()['study_accession'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>size_frac</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0.8-5.00</th>\n",
       "      <td>316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180-2000.00</th>\n",
       "      <td>183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20-180.00</th>\n",
       "      <td>178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5-20.00</th>\n",
       "      <td>164</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             study_accession\n",
       "size_frac                   \n",
       "0.8-5.00                 316\n",
       "180-2000.00              183\n",
       "20-180.00                178\n",
       "5-20.00                  164"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#drop 0.22 sample\n",
    "metat = metat.replace('0.8->0.80', '0.8-5.00')\n",
    "metat = metat.replace('0.8-20.00', '0.8-5.00')\n",
    "metat = metat.replace('0.8-3.00', '0.8-5.00')\n",
    "metat = metat.replace('3.0->3.00', '0.8-5.00')\n",
    "metat = metat.replace('3.0->5.00', '0.8-5.00')\n",
    "metat = metat.replace('180.0-2000.00', '180-2000.00')\n",
    "metat = metat.replace('20.0-180.00', '20-180.00')\n",
    "metat = metat.replace('5.0-20.00', '5-20.00')\n",
    "\n",
    "\n",
    "pd.DataFrame(metat.groupby('size_frac').count()['study_accession'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check Ocean Region and Depth and clean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "metag['OS'] = metag['OS region'].str.split(' ').str[0].str.strip('[').str.strip(']')\n",
    "metag['EF'] = metag['Env feature'].str.split(' ').str[0].str.strip('[').str.strip(']')\n",
    "metag['BG'] = metag['BG province'].str.split(' ').str[0].str.strip('[').str.strip(']')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "metat['OS'] = metat['OS region'].str.split(' ').str[0].str.strip('[').str.strip(']')\n",
    "metat['EF'] = metat['Env feature'].str.split(' ').str[0].str.strip('[').str.strip(']')\n",
    "metat['BG'] = metat['BG province'].str.split(' ').str[0].str.strip('[').str.strip(']')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "subset = ['study_accession', 'run_accession', 'Station', 'size_frac', 'Env feature', \n",
    "        'Sample material', 'Depth, nominal', 'OS region', 'BG province',\n",
    "        'Latitude', 'Longitude', 'OS', 'EF', 'BG']\n",
    "\n",
    "metag_subset = metag[subset]\n",
    "metat_subset = metat[subset]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Function to split into groups (for assembly)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "def split_assembly_groups(df, OS_list):\n",
    "    df['ERR_list']= df.groupby(['OS', 'EF', 'size_frac'])['run_accession'].transform(lambda x: ', '.join(x))\n",
    "    df_combo = df[['OS', 'EF', 'size_frac','ERR_list']].drop_duplicates()\n",
    "    df = df.drop('ERR_list', axis =1)\n",
    "    for O in OS_list:\n",
    "        df_combo = df_combo[df_combo.OS != O]\n",
    "        tmp = df[df.OS == O]\n",
    "        tmp['ERR_list']=tmp.groupby(['OS','BG', 'EF', 'size_frac'])['run_accession'].transform(lambda x: ', '.join(x))\n",
    "        tmp_combo =tmp[['OS', 'BG', 'EF', 'size_frac','ERR_list']].drop_duplicates()\n",
    "    final_combo = pd.concat([df_combo, tmp_combo])  \n",
    "    final_combo = final_combo.fillna('all')\n",
    "    final_combo['Assembly_group'] = final_combo.OS + '-' + final_combo.BG + '-' + final_combo.EF + '-' +final_combo.size_frac\n",
    "    final_combo['Sub_region'] = final_combo.OS + '-' + final_combo.BG\n",
    "    final_combo['Depth_sizefrac'] = final_combo.EF + '-' +final_combo.size_frac\n",
    "    final_combo['ERR_count']= final_combo['ERR_list'].str.split(', ').str.len()\n",
    "    final_combo = final_combo[['Sub_region', 'Depth_sizefrac', 'ERR_count', 'ERR_list', 'Assembly_group']]\n",
    "    final_combo = final_combo.set_index('Sub_region')\n",
    "    return(final_combo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/vortexfs1/home/sarahhu/.conda/envs/r_3.5.1/lib/python3.6/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/vortexfs1/home/sarahhu/.conda/envs/r_3.5.1/lib/python3.6/site-packages/ipykernel_launcher.py:10: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  # Remove the CWD from sys.path while we load stuff.\n",
      "/vortexfs1/home/sarahhu/.conda/envs/r_3.5.1/lib/python3.6/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/vortexfs1/home/sarahhu/.conda/envs/r_3.5.1/lib/python3.6/site-packages/ipykernel_launcher.py:10: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  # Remove the CWD from sys.path while we load stuff.\n"
     ]
    }
   ],
   "source": [
    "OS_list = ['SPO']\n",
    "metag_out = split_assembly_groups(metag, OS_list)\n",
    "metat_out = split_assembly_groups(metat, OS_list)\n",
    "\n",
    "metag_out.to_csv('SampleList_ForAssembly_metaG_python.txt', sep = '\\t')\n",
    "metat_out.to_csv('SampleList_ForAssembly_metaT_python.txt', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Depth_sizefrac</th>\n",
       "      <th>ERR_count</th>\n",
       "      <th>ERR_list</th>\n",
       "      <th>Assembly_group</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sub_region</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>MES-0.8-5.00</td>\n",
       "      <td>1</td>\n",
       "      <td>ERR868377</td>\n",
       "      <td>SPO-CHIL-MES-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>MES-0.8-5.00</td>\n",
       "      <td>1</td>\n",
       "      <td>ERR868362</td>\n",
       "      <td>SPO-PEOD-MES-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>DCM-20-180.00</td>\n",
       "      <td>1</td>\n",
       "      <td>ERR1726830</td>\n",
       "      <td>SPO-CHIL-DCM-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RS-all</th>\n",
       "      <td>DCM-5-20.00</td>\n",
       "      <td>1</td>\n",
       "      <td>ERR1726910</td>\n",
       "      <td>RS-all-DCM-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SO-all</th>\n",
       "      <td>DCM-20-180.00</td>\n",
       "      <td>1</td>\n",
       "      <td>ERR599288</td>\n",
       "      <td>SO-all-DCM-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>DCM-5-20.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1700901, ERR1726828</td>\n",
       "      <td>SPO-CHIL-DCM-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>MIX-5-20.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726545, ERR1726756</td>\n",
       "      <td>NAO-all-MIX-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>DCM-180-2000.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1700911, ERR1726962</td>\n",
       "      <td>SPO-PEOD-DCM-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>MIX-180-2000.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726532, ERR1726941</td>\n",
       "      <td>NAO-all-MIX-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>DCM-5-20.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726562, ERR1726742</td>\n",
       "      <td>SPO-PEOD-DCM-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SO-all</th>\n",
       "      <td>MES-0.8-5.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR599217, ERR599227</td>\n",
       "      <td>SO-all-MES-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>DCM-180-2000.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726566, ERR1726627</td>\n",
       "      <td>SPO-CHIL-DCM-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726577, ERR1726709</td>\n",
       "      <td>SPO-PEOD-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726594, ERR1726841</td>\n",
       "      <td>SPO-PEOD-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SO-all</th>\n",
       "      <td>DCM-5-20.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1740327, ERR599206</td>\n",
       "      <td>SO-all-DCM-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>DCM-20-180.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726659, ERR1726727</td>\n",
       "      <td>SPO-PEOD-DCM-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>SRF-180-2000.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726705, ERR1726795</td>\n",
       "      <td>SPO-PEOD-SRF-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SO-all</th>\n",
       "      <td>DCM-180-2000.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726812, ERR599310</td>\n",
       "      <td>SO-all-DCM-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>DCM-0.8-5.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR868358, ERR868505</td>\n",
       "      <td>SPO-CHIL-DCM-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RS-all</th>\n",
       "      <td>DCM-0.8-5.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR868348, ERR868439</td>\n",
       "      <td>RS-all-DCM-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>FSW-0.8-5.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR868368, ERR868484</td>\n",
       "      <td>IO-all-FSW-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RS-all</th>\n",
       "      <td>DCM-20-180.00</td>\n",
       "      <td>3</td>\n",
       "      <td>ERR1726568, ERR1726610, ERR1726960</td>\n",
       "      <td>RS-all-DCM-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SO-all</th>\n",
       "      <td>DCM-0.8-5.00</td>\n",
       "      <td>3</td>\n",
       "      <td>ERR599239, ERR599325, ERR868361</td>\n",
       "      <td>SO-all-DCM-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RS-all</th>\n",
       "      <td>DCM-180-2000.00</td>\n",
       "      <td>3</td>\n",
       "      <td>ERR1726537, ERR1726578, ERR1726905</td>\n",
       "      <td>RS-all-DCM-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>MIX-20-180.00</td>\n",
       "      <td>3</td>\n",
       "      <td>ERR1726653, ERR1726803, ERR1726947</td>\n",
       "      <td>NAO-all-MIX-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>3</td>\n",
       "      <td>ERR1726632, ERR1726713, ERR1726878</td>\n",
       "      <td>SPO-CHIL-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>4</td>\n",
       "      <td>ERR1726524, ERR1726525, ERR1726850, ERR1726952</td>\n",
       "      <td>SPO-CHIL-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>DCM-20-180.00</td>\n",
       "      <td>4</td>\n",
       "      <td>ERR1726529, ERR1726852, ERR1726893, ERR1726911</td>\n",
       "      <td>NAO-all-DCM-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>SRF-0.8-5.00</td>\n",
       "      <td>4</td>\n",
       "      <td>ERR1726628, ERR1726851, ERR868357, ERR868462</td>\n",
       "      <td>SPO-PEOD-SRF-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SO-all</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>4</td>\n",
       "      <td>ERR1726939, ERR1726968, ERR599224, ERR599228</td>\n",
       "      <td>SO-all-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NPO-all</th>\n",
       "      <td>DCM-20-180.00</td>\n",
       "      <td>11</td>\n",
       "      <td>ERR1726563, ERR1726641, ERR1726647, ERR1726658...</td>\n",
       "      <td>NPO-all-DCM-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>11</td>\n",
       "      <td>ERR1726581, ERR1726625, ERR1726635, ERR1726720...</td>\n",
       "      <td>NAO-all-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>11</td>\n",
       "      <td>ERR1700893, ERR1700902, ERR1726534, ERR1726542...</td>\n",
       "      <td>NAO-all-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>DCM-180-2000.00</td>\n",
       "      <td>12</td>\n",
       "      <td>ERR1726526, ERR1726686, ERR1726696, ERR1726746...</td>\n",
       "      <td>IO-all-DCM-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>DCM-180-2000.00</td>\n",
       "      <td>12</td>\n",
       "      <td>ERR1726550, ERR1726552, ERR1726604, ERR1726623...</td>\n",
       "      <td>SPO-SPSG-DCM-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NPO-all</th>\n",
       "      <td>SRF-180-2000.00</td>\n",
       "      <td>12</td>\n",
       "      <td>ERR1726536, ERR1726619, ERR1726738, ERR1726747...</td>\n",
       "      <td>NPO-all-SRF-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAO-all</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>12</td>\n",
       "      <td>ERR1700898, ERR1726626, ERR1726881, ERR1726969...</td>\n",
       "      <td>SAO-all-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>12</td>\n",
       "      <td>ERR1700890, ERR1700899, ERR1726575, ERR1726677...</td>\n",
       "      <td>IO-all-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MS-all</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>13</td>\n",
       "      <td>ERR1726546, ERR1726613, ERR1726642, ERR1726660...</td>\n",
       "      <td>MS-all-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>DCM-0.8-5.00</td>\n",
       "      <td>13</td>\n",
       "      <td>ERR1726633, ERR1726897, ERR599323, ERR599324, ...</td>\n",
       "      <td>IO-all-DCM-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>SRF-180-2000.00</td>\n",
       "      <td>13</td>\n",
       "      <td>ERR1726553, ERR1726638, ERR1726654, ERR1726712...</td>\n",
       "      <td>NAO-all-SRF-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAO-all</th>\n",
       "      <td>SRF-180-2000.00</td>\n",
       "      <td>13</td>\n",
       "      <td>ERR1726678, ERR1726689, ERR1726819, ERR1726928...</td>\n",
       "      <td>SAO-all-SRF-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NPO-all</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>13</td>\n",
       "      <td>ERR1700910, ERR1726559, ERR1726640, ERR1726669...</td>\n",
       "      <td>NPO-all-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NPO-all</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>13</td>\n",
       "      <td>ERR1726580, ERR1726621, ERR1726661, ERR1726687...</td>\n",
       "      <td>NPO-all-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAO-all</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>14</td>\n",
       "      <td>ERR1726636, ERR1726663, ERR1726731, ERR1726853...</td>\n",
       "      <td>SAO-all-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MS-all</th>\n",
       "      <td>SRF-0.8-5.00</td>\n",
       "      <td>14</td>\n",
       "      <td>ERR1726597, ERR538173, ERR538186, ERR868356, E...</td>\n",
       "      <td>MS-all-SRF-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NPO-all</th>\n",
       "      <td>DCM-0.8-5.00</td>\n",
       "      <td>15</td>\n",
       "      <td>ERR1726523, ERR1726570, ERR1726724, ERR1726760...</td>\n",
       "      <td>NPO-all-DCM-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MS-all</th>\n",
       "      <td>SRF-180-2000.00</td>\n",
       "      <td>15</td>\n",
       "      <td>ERR1726574, ERR1726612, ERR1726662, ERR1726681...</td>\n",
       "      <td>MS-all-SRF-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>17</td>\n",
       "      <td>ERR1700914, ERR1726587, ERR1726599, ERR1726611...</td>\n",
       "      <td>IO-all-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>18</td>\n",
       "      <td>ERR1726522, ERR1726533, ERR1726535, ERR1726571...</td>\n",
       "      <td>SPO-SPSG-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>DCM-0.8-5.00</td>\n",
       "      <td>19</td>\n",
       "      <td>ERR1726527, ERR1726579, ERR1726584, ERR1726609...</td>\n",
       "      <td>SPO-SPSG-DCM-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>SRF-0.8-5.00</td>\n",
       "      <td>20</td>\n",
       "      <td>ERR1726549, ERR1740328, ERR599210, ERR599266, ...</td>\n",
       "      <td>IO-all-SRF-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NPO-all</th>\n",
       "      <td>SRF-0.8-5.00</td>\n",
       "      <td>20</td>\n",
       "      <td>ERR1726576, ERR1726730, ERR1726807, ERR1726816...</td>\n",
       "      <td>NPO-all-SRF-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>SRF-0.8-5.00</td>\n",
       "      <td>20</td>\n",
       "      <td>ERR1726589, ERR1726676, ERR1726679, ERR1726693...</td>\n",
       "      <td>NAO-all-SRF-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>SRF-180-2000.00</td>\n",
       "      <td>21</td>\n",
       "      <td>ERR1700889, ERR1700906, ERR1700907, ERR1700909...</td>\n",
       "      <td>IO-all-SRF-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAO-all</th>\n",
       "      <td>SRF-0.8-5.00</td>\n",
       "      <td>23</td>\n",
       "      <td>ERR1726572, ERR1726699, ERR1726733, ERR1726975...</td>\n",
       "      <td>SAO-all-SRF-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>25</td>\n",
       "      <td>ERR1726528, ERR1726544, ERR1726558, ERR1726588...</td>\n",
       "      <td>SPO-SPSG-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>SRF-180-2000.00</td>\n",
       "      <td>27</td>\n",
       "      <td>ERR1726541, ERR1726543, ERR1726561, ERR1726586...</td>\n",
       "      <td>SPO-SPSG-SRF-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>SRF-0.8-5.00</td>\n",
       "      <td>33</td>\n",
       "      <td>ERR1726556, ERR1726564, ERR1726567, ERR1726569...</td>\n",
       "      <td>SPO-SPSG-SRF-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAO-all</th>\n",
       "      <td>DCM-0.8-5.00</td>\n",
       "      <td>40</td>\n",
       "      <td>ERR1726682, ERR1726701, ERR1726704, ERR1726793...</td>\n",
       "      <td>SAO-all-DCM-0.8-5.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>94 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Depth_sizefrac  ERR_count  \\\n",
       "Sub_region                               \n",
       "SPO-CHIL       MES-0.8-5.00          1   \n",
       "SPO-PEOD       MES-0.8-5.00          1   \n",
       "SPO-CHIL      DCM-20-180.00          1   \n",
       "RS-all          DCM-5-20.00          1   \n",
       "SO-all        DCM-20-180.00          1   \n",
       "SPO-CHIL        DCM-5-20.00          2   \n",
       "NAO-all         MIX-5-20.00          2   \n",
       "SPO-PEOD    DCM-180-2000.00          2   \n",
       "NAO-all     MIX-180-2000.00          2   \n",
       "SPO-PEOD        DCM-5-20.00          2   \n",
       "SO-all         MES-0.8-5.00          2   \n",
       "SPO-CHIL    DCM-180-2000.00          2   \n",
       "SPO-PEOD      SRF-20-180.00          2   \n",
       "SPO-PEOD        SRF-5-20.00          2   \n",
       "SO-all          DCM-5-20.00          2   \n",
       "SPO-PEOD      DCM-20-180.00          2   \n",
       "SPO-PEOD    SRF-180-2000.00          2   \n",
       "SO-all      DCM-180-2000.00          2   \n",
       "SPO-CHIL       DCM-0.8-5.00          2   \n",
       "RS-all         DCM-0.8-5.00          2   \n",
       "IO-all         FSW-0.8-5.00          2   \n",
       "RS-all        DCM-20-180.00          3   \n",
       "SO-all         DCM-0.8-5.00          3   \n",
       "RS-all      DCM-180-2000.00          3   \n",
       "NAO-all       MIX-20-180.00          3   \n",
       "SPO-CHIL        SRF-5-20.00          3   \n",
       "SPO-CHIL      SRF-20-180.00          4   \n",
       "NAO-all       DCM-20-180.00          4   \n",
       "SPO-PEOD       SRF-0.8-5.00          4   \n",
       "SO-all          SRF-5-20.00          4   \n",
       "...                     ...        ...   \n",
       "NPO-all       DCM-20-180.00         11   \n",
       "NAO-all       SRF-20-180.00         11   \n",
       "NAO-all         SRF-5-20.00         11   \n",
       "IO-all      DCM-180-2000.00         12   \n",
       "SPO-SPSG    DCM-180-2000.00         12   \n",
       "NPO-all     SRF-180-2000.00         12   \n",
       "SAO-all         SRF-5-20.00         12   \n",
       "IO-all          SRF-5-20.00         12   \n",
       "MS-all        SRF-20-180.00         13   \n",
       "IO-all         DCM-0.8-5.00         13   \n",
       "NAO-all     SRF-180-2000.00         13   \n",
       "SAO-all     SRF-180-2000.00         13   \n",
       "NPO-all         SRF-5-20.00         13   \n",
       "NPO-all       SRF-20-180.00         13   \n",
       "SAO-all       SRF-20-180.00         14   \n",
       "MS-all         SRF-0.8-5.00         14   \n",
       "NPO-all        DCM-0.8-5.00         15   \n",
       "MS-all      SRF-180-2000.00         15   \n",
       "IO-all        SRF-20-180.00         17   \n",
       "SPO-SPSG        SRF-5-20.00         18   \n",
       "SPO-SPSG       DCM-0.8-5.00         19   \n",
       "IO-all         SRF-0.8-5.00         20   \n",
       "NPO-all        SRF-0.8-5.00         20   \n",
       "NAO-all        SRF-0.8-5.00         20   \n",
       "IO-all      SRF-180-2000.00         21   \n",
       "SAO-all        SRF-0.8-5.00         23   \n",
       "SPO-SPSG      SRF-20-180.00         25   \n",
       "SPO-SPSG    SRF-180-2000.00         27   \n",
       "SPO-SPSG       SRF-0.8-5.00         33   \n",
       "SAO-all        DCM-0.8-5.00         40   \n",
       "\n",
       "                                                     ERR_list  \\\n",
       "Sub_region                                                      \n",
       "SPO-CHIL                                            ERR868377   \n",
       "SPO-PEOD                                            ERR868362   \n",
       "SPO-CHIL                                           ERR1726830   \n",
       "RS-all                                             ERR1726910   \n",
       "SO-all                                              ERR599288   \n",
       "SPO-CHIL                               ERR1700901, ERR1726828   \n",
       "NAO-all                                ERR1726545, ERR1726756   \n",
       "SPO-PEOD                               ERR1700911, ERR1726962   \n",
       "NAO-all                                ERR1726532, ERR1726941   \n",
       "SPO-PEOD                               ERR1726562, ERR1726742   \n",
       "SO-all                                   ERR599217, ERR599227   \n",
       "SPO-CHIL                               ERR1726566, ERR1726627   \n",
       "SPO-PEOD                               ERR1726577, ERR1726709   \n",
       "SPO-PEOD                               ERR1726594, ERR1726841   \n",
       "SO-all                                  ERR1740327, ERR599206   \n",
       "SPO-PEOD                               ERR1726659, ERR1726727   \n",
       "SPO-PEOD                               ERR1726705, ERR1726795   \n",
       "SO-all                                  ERR1726812, ERR599310   \n",
       "SPO-CHIL                                 ERR868358, ERR868505   \n",
       "RS-all                                   ERR868348, ERR868439   \n",
       "IO-all                                   ERR868368, ERR868484   \n",
       "RS-all                     ERR1726568, ERR1726610, ERR1726960   \n",
       "SO-all                        ERR599239, ERR599325, ERR868361   \n",
       "RS-all                     ERR1726537, ERR1726578, ERR1726905   \n",
       "NAO-all                    ERR1726653, ERR1726803, ERR1726947   \n",
       "SPO-CHIL                   ERR1726632, ERR1726713, ERR1726878   \n",
       "SPO-CHIL       ERR1726524, ERR1726525, ERR1726850, ERR1726952   \n",
       "NAO-all        ERR1726529, ERR1726852, ERR1726893, ERR1726911   \n",
       "SPO-PEOD         ERR1726628, ERR1726851, ERR868357, ERR868462   \n",
       "SO-all           ERR1726939, ERR1726968, ERR599224, ERR599228   \n",
       "...                                                       ...   \n",
       "NPO-all     ERR1726563, ERR1726641, ERR1726647, ERR1726658...   \n",
       "NAO-all     ERR1726581, ERR1726625, ERR1726635, ERR1726720...   \n",
       "NAO-all     ERR1700893, ERR1700902, ERR1726534, ERR1726542...   \n",
       "IO-all      ERR1726526, ERR1726686, ERR1726696, ERR1726746...   \n",
       "SPO-SPSG    ERR1726550, ERR1726552, ERR1726604, ERR1726623...   \n",
       "NPO-all     ERR1726536, ERR1726619, ERR1726738, ERR1726747...   \n",
       "SAO-all     ERR1700898, ERR1726626, ERR1726881, ERR1726969...   \n",
       "IO-all      ERR1700890, ERR1700899, ERR1726575, ERR1726677...   \n",
       "MS-all      ERR1726546, ERR1726613, ERR1726642, ERR1726660...   \n",
       "IO-all      ERR1726633, ERR1726897, ERR599323, ERR599324, ...   \n",
       "NAO-all     ERR1726553, ERR1726638, ERR1726654, ERR1726712...   \n",
       "SAO-all     ERR1726678, ERR1726689, ERR1726819, ERR1726928...   \n",
       "NPO-all     ERR1700910, ERR1726559, ERR1726640, ERR1726669...   \n",
       "NPO-all     ERR1726580, ERR1726621, ERR1726661, ERR1726687...   \n",
       "SAO-all     ERR1726636, ERR1726663, ERR1726731, ERR1726853...   \n",
       "MS-all      ERR1726597, ERR538173, ERR538186, ERR868356, E...   \n",
       "NPO-all     ERR1726523, ERR1726570, ERR1726724, ERR1726760...   \n",
       "MS-all      ERR1726574, ERR1726612, ERR1726662, ERR1726681...   \n",
       "IO-all      ERR1700914, ERR1726587, ERR1726599, ERR1726611...   \n",
       "SPO-SPSG    ERR1726522, ERR1726533, ERR1726535, ERR1726571...   \n",
       "SPO-SPSG    ERR1726527, ERR1726579, ERR1726584, ERR1726609...   \n",
       "IO-all      ERR1726549, ERR1740328, ERR599210, ERR599266, ...   \n",
       "NPO-all     ERR1726576, ERR1726730, ERR1726807, ERR1726816...   \n",
       "NAO-all     ERR1726589, ERR1726676, ERR1726679, ERR1726693...   \n",
       "IO-all      ERR1700889, ERR1700906, ERR1700907, ERR1700909...   \n",
       "SAO-all     ERR1726572, ERR1726699, ERR1726733, ERR1726975...   \n",
       "SPO-SPSG    ERR1726528, ERR1726544, ERR1726558, ERR1726588...   \n",
       "SPO-SPSG    ERR1726541, ERR1726543, ERR1726561, ERR1726586...   \n",
       "SPO-SPSG    ERR1726556, ERR1726564, ERR1726567, ERR1726569...   \n",
       "SAO-all     ERR1726682, ERR1726701, ERR1726704, ERR1726793...   \n",
       "\n",
       "                      Assembly_group  \n",
       "Sub_region                            \n",
       "SPO-CHIL       SPO-CHIL-MES-0.8-5.00  \n",
       "SPO-PEOD       SPO-PEOD-MES-0.8-5.00  \n",
       "SPO-CHIL      SPO-CHIL-DCM-20-180.00  \n",
       "RS-all            RS-all-DCM-5-20.00  \n",
       "SO-all          SO-all-DCM-20-180.00  \n",
       "SPO-CHIL        SPO-CHIL-DCM-5-20.00  \n",
       "NAO-all          NAO-all-MIX-5-20.00  \n",
       "SPO-PEOD    SPO-PEOD-DCM-180-2000.00  \n",
       "NAO-all      NAO-all-MIX-180-2000.00  \n",
       "SPO-PEOD        SPO-PEOD-DCM-5-20.00  \n",
       "SO-all           SO-all-MES-0.8-5.00  \n",
       "SPO-CHIL    SPO-CHIL-DCM-180-2000.00  \n",
       "SPO-PEOD      SPO-PEOD-SRF-20-180.00  \n",
       "SPO-PEOD        SPO-PEOD-SRF-5-20.00  \n",
       "SO-all            SO-all-DCM-5-20.00  \n",
       "SPO-PEOD      SPO-PEOD-DCM-20-180.00  \n",
       "SPO-PEOD    SPO-PEOD-SRF-180-2000.00  \n",
       "SO-all        SO-all-DCM-180-2000.00  \n",
       "SPO-CHIL       SPO-CHIL-DCM-0.8-5.00  \n",
       "RS-all           RS-all-DCM-0.8-5.00  \n",
       "IO-all           IO-all-FSW-0.8-5.00  \n",
       "RS-all          RS-all-DCM-20-180.00  \n",
       "SO-all           SO-all-DCM-0.8-5.00  \n",
       "RS-all        RS-all-DCM-180-2000.00  \n",
       "NAO-all        NAO-all-MIX-20-180.00  \n",
       "SPO-CHIL        SPO-CHIL-SRF-5-20.00  \n",
       "SPO-CHIL      SPO-CHIL-SRF-20-180.00  \n",
       "NAO-all        NAO-all-DCM-20-180.00  \n",
       "SPO-PEOD       SPO-PEOD-SRF-0.8-5.00  \n",
       "SO-all            SO-all-SRF-5-20.00  \n",
       "...                              ...  \n",
       "NPO-all        NPO-all-DCM-20-180.00  \n",
       "NAO-all        NAO-all-SRF-20-180.00  \n",
       "NAO-all          NAO-all-SRF-5-20.00  \n",
       "IO-all        IO-all-DCM-180-2000.00  \n",
       "SPO-SPSG    SPO-SPSG-DCM-180-2000.00  \n",
       "NPO-all      NPO-all-SRF-180-2000.00  \n",
       "SAO-all          SAO-all-SRF-5-20.00  \n",
       "IO-all            IO-all-SRF-5-20.00  \n",
       "MS-all          MS-all-SRF-20-180.00  \n",
       "IO-all           IO-all-DCM-0.8-5.00  \n",
       "NAO-all      NAO-all-SRF-180-2000.00  \n",
       "SAO-all      SAO-all-SRF-180-2000.00  \n",
       "NPO-all          NPO-all-SRF-5-20.00  \n",
       "NPO-all        NPO-all-SRF-20-180.00  \n",
       "SAO-all        SAO-all-SRF-20-180.00  \n",
       "MS-all           MS-all-SRF-0.8-5.00  \n",
       "NPO-all         NPO-all-DCM-0.8-5.00  \n",
       "MS-all        MS-all-SRF-180-2000.00  \n",
       "IO-all          IO-all-SRF-20-180.00  \n",
       "SPO-SPSG        SPO-SPSG-SRF-5-20.00  \n",
       "SPO-SPSG       SPO-SPSG-DCM-0.8-5.00  \n",
       "IO-all           IO-all-SRF-0.8-5.00  \n",
       "NPO-all         NPO-all-SRF-0.8-5.00  \n",
       "NAO-all         NAO-all-SRF-0.8-5.00  \n",
       "IO-all        IO-all-SRF-180-2000.00  \n",
       "SAO-all         SAO-all-SRF-0.8-5.00  \n",
       "SPO-SPSG      SPO-SPSG-SRF-20-180.00  \n",
       "SPO-SPSG    SPO-SPSG-SRF-180-2000.00  \n",
       "SPO-SPSG       SPO-SPSG-SRF-0.8-5.00  \n",
       "SAO-all         SAO-all-DCM-0.8-5.00  \n",
       "\n",
       "[94 rows x 4 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metag_out.sort_values(['ERR_count'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate test data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>study_accession</th>\n",
       "      <th>sample_accession</th>\n",
       "      <th>secondary_sample_accession</th>\n",
       "      <th>experiment_accession</th>\n",
       "      <th>run_accession</th>\n",
       "      <th>tax_id</th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>...</th>\n",
       "      <th>u [cm/s]</th>\n",
       "      <th>v [cm/s]</th>\n",
       "      <th>OW</th>\n",
       "      <th>MLE [1/day]</th>\n",
       "      <th>RT [days]</th>\n",
       "      <th>Latitude (of the continental shelf with...)</th>\n",
       "      <th>Longitude (of the continental shelf with...)</th>\n",
       "      <th>Age [days]</th>\n",
       "      <th>Latitude (of the geographic coordinate ...)</th>\n",
       "      <th>Longitude (of the geographic coordinate ...)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2656965</td>\n",
       "      <td>ERS488900</td>\n",
       "      <td>ERX1771177</td>\n",
       "      <td>ERR1700889</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.557961</td>\n",
       "      <td>15.046586</td>\n",
       "      <td>0.403157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.250000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>16.583327</td>\n",
       "      <td>63.625415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620205</td>\n",
       "      <td>ERS489054</td>\n",
       "      <td>ERX1771189</td>\n",
       "      <td>ERR1700890</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.466550</td>\n",
       "      <td>4.306080</td>\n",
       "      <td>-0.412443</td>\n",
       "      <td>0.026178</td>\n",
       "      <td>19.027778</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>14.404658</td>\n",
       "      <td>69.689377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623506</td>\n",
       "      <td>ERS493999</td>\n",
       "      <td>ERX1771187</td>\n",
       "      <td>ERR1700891</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>28.262368</td>\n",
       "      <td>8.317923</td>\n",
       "      <td>-2.922481</td>\n",
       "      <td>0.044572</td>\n",
       "      <td>25.750000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>26.338683</td>\n",
       "      <td>-87.272865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620172</td>\n",
       "      <td>ERS489013</td>\n",
       "      <td>ERX1771191</td>\n",
       "      <td>ERR1700892</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>33.111660</td>\n",
       "      <td>-33.418508</td>\n",
       "      <td>0.043242</td>\n",
       "      <td>0.086921</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>17.386625</td>\n",
       "      <td>66.210146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2730950</td>\n",
       "      <td>ERS494049</td>\n",
       "      <td>ERX1771202</td>\n",
       "      <td>ERR1700893</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>8.319742</td>\n",
       "      <td>91.787857</td>\n",
       "      <td>-4.707638</td>\n",
       "      <td>0.398799</td>\n",
       "      <td>1.083333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619554</td>\n",
       "      <td>ERS488153</td>\n",
       "      <td>ERX1771192</td>\n",
       "      <td>ERR1700894</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-16.430919</td>\n",
       "      <td>4.368810</td>\n",
       "      <td>0.021447</td>\n",
       "      <td>0.190651</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>38.821749</td>\n",
       "      <td>6.902024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619623</td>\n",
       "      <td>ERS488251</td>\n",
       "      <td>ERX1771178</td>\n",
       "      <td>ERR1700895</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.797800</td>\n",
       "      <td>41.665900</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>7</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619725</td>\n",
       "      <td>ERS488412</td>\n",
       "      <td>ERX1771195</td>\n",
       "      <td>ERR1700896</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-5.127640</td>\n",
       "      <td>-0.375367</td>\n",
       "      <td>0.051438</td>\n",
       "      <td>0.087535</td>\n",
       "      <td>1.333333</td>\n",
       "      <td>14.973150</td>\n",
       "      <td>34.451200</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32.144171</td>\n",
       "      <td>16.847326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>8</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622039</td>\n",
       "      <td>ERS491758</td>\n",
       "      <td>ERX1771199</td>\n",
       "      <td>ERR1700897</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>14.212952</td>\n",
       "      <td>1.275138</td>\n",
       "      <td>-0.006839</td>\n",
       "      <td>0.028784</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-25.231969</td>\n",
       "      <td>-112.818256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2621474</td>\n",
       "      <td>ERS490981</td>\n",
       "      <td>ERX1771201</td>\n",
       "      <td>ERR1700898</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>10.611410</td>\n",
       "      <td>1.423920</td>\n",
       "      <td>0.072135</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-65.022700</td>\n",
       "      <td>-54.418000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620072</td>\n",
       "      <td>ERS488886</td>\n",
       "      <td>ERX1771182</td>\n",
       "      <td>ERR1700899</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.557961</td>\n",
       "      <td>15.046586</td>\n",
       "      <td>0.403157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.250000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>16.583327</td>\n",
       "      <td>63.625415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>11</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619421</td>\n",
       "      <td>ERS487958</td>\n",
       "      <td>ERX1771180</td>\n",
       "      <td>ERR1700900</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.031746</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>-6.552800</td>\n",
       "      <td>36.563150</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>12</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2621827</td>\n",
       "      <td>ERS491478</td>\n",
       "      <td>ERX1771186</td>\n",
       "      <td>ERR1700901</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>4.738948</td>\n",
       "      <td>-4.678504</td>\n",
       "      <td>0.061918</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.250000</td>\n",
       "      <td>-72.642651</td>\n",
       "      <td>-35.181114</td>\n",
       "      <td>63.0</td>\n",
       "      <td>-34.302478</td>\n",
       "      <td>-73.496846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>13</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2731011</td>\n",
       "      <td>ERS494136</td>\n",
       "      <td>ERX1771181</td>\n",
       "      <td>ERR1700902</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>47.329031</td>\n",
       "      <td>34.457153</td>\n",
       "      <td>2.028463</td>\n",
       "      <td>0.249201</td>\n",
       "      <td>1.861111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>29.041205</td>\n",
       "      <td>-79.321465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>14</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622132</td>\n",
       "      <td>ERS491887</td>\n",
       "      <td>ERX1771196</td>\n",
       "      <td>ERR1700903</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.841770</td>\n",
       "      <td>2.208339</td>\n",
       "      <td>0.013354</td>\n",
       "      <td>0.045278</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-13.259274</td>\n",
       "      <td>-94.231260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>15</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619959</td>\n",
       "      <td>ERS488754</td>\n",
       "      <td>ERX1771197</td>\n",
       "      <td>ERR1700904</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>3.924148</td>\n",
       "      <td>8.325847</td>\n",
       "      <td>-0.030610</td>\n",
       "      <td>0.074837</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>23.584126</td>\n",
       "      <td>60.241719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>16</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622039</td>\n",
       "      <td>ERS491758</td>\n",
       "      <td>ERX1771185</td>\n",
       "      <td>ERR1700905</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>14.212952</td>\n",
       "      <td>1.275138</td>\n",
       "      <td>-0.006839</td>\n",
       "      <td>0.028784</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-25.231969</td>\n",
       "      <td>-112.818256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>17</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2657022</td>\n",
       "      <td>ERS488820</td>\n",
       "      <td>ERX1771190</td>\n",
       "      <td>ERR1700906</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>10.755607</td>\n",
       "      <td>-6.652039</td>\n",
       "      <td>0.005785</td>\n",
       "      <td>0.069484</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>20.258041</td>\n",
       "      <td>64.562537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>18</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2657022</td>\n",
       "      <td>ERS488820</td>\n",
       "      <td>ERX1771188</td>\n",
       "      <td>ERR1700907</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>10.755607</td>\n",
       "      <td>-6.652039</td>\n",
       "      <td>0.005785</td>\n",
       "      <td>0.069484</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>20.258041</td>\n",
       "      <td>64.562537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>19</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619421</td>\n",
       "      <td>ERS487958</td>\n",
       "      <td>ERX1771198</td>\n",
       "      <td>ERR1700908</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.031746</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>-6.552800</td>\n",
       "      <td>36.563150</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>20</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2657050</td>\n",
       "      <td>ERS489336</td>\n",
       "      <td>ERX1771193</td>\n",
       "      <td>ERR1700909</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>3.682729</td>\n",
       "      <td>19.395807</td>\n",
       "      <td>-0.118280</td>\n",
       "      <td>0.067544</td>\n",
       "      <td>3.777778</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-14.979369</td>\n",
       "      <td>71.710551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>21</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2732453</td>\n",
       "      <td>ERS493776</td>\n",
       "      <td>ERX1771200</td>\n",
       "      <td>ERR1700910</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>52.020196</td>\n",
       "      <td>1.040314</td>\n",
       "      <td>-0.002795</td>\n",
       "      <td>0.020154</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>7.512087</td>\n",
       "      <td>-111.603363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>22</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2657097</td>\n",
       "      <td>ERS492000</td>\n",
       "      <td>ERX1771184</td>\n",
       "      <td>ERR1700911</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-12.746951</td>\n",
       "      <td>15.658745</td>\n",
       "      <td>0.043188</td>\n",
       "      <td>0.012324</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-9.444510</td>\n",
       "      <td>-86.887739</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>23</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620222</td>\n",
       "      <td>ERS489079</td>\n",
       "      <td>ERX1771194</td>\n",
       "      <td>ERR1700912</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.466550</td>\n",
       "      <td>4.306080</td>\n",
       "      <td>-0.412443</td>\n",
       "      <td>0.026178</td>\n",
       "      <td>19.027778</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>14.404658</td>\n",
       "      <td>69.689377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>24</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619789</td>\n",
       "      <td>ERS488516</td>\n",
       "      <td>ERX1771183</td>\n",
       "      <td>ERR1700913</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.449987</td>\n",
       "      <td>5.586469</td>\n",
       "      <td>0.369749</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>25</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2656981</td>\n",
       "      <td>ERS488816</td>\n",
       "      <td>ERX1771179</td>\n",
       "      <td>ERR1700914</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>10.755607</td>\n",
       "      <td>-6.652039</td>\n",
       "      <td>0.005785</td>\n",
       "      <td>0.069484</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>20.258041</td>\n",
       "      <td>64.562537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>26</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2657109</td>\n",
       "      <td>ERS492751</td>\n",
       "      <td>ERX1796605</td>\n",
       "      <td>ERR1726522</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.286042</td>\n",
       "      <td>2.890793</td>\n",
       "      <td>-0.011503</td>\n",
       "      <td>0.053774</td>\n",
       "      <td>1.083333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-8.878592</td>\n",
       "      <td>-137.774399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>27</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623302</td>\n",
       "      <td>ERS493677</td>\n",
       "      <td>ERX1796551</td>\n",
       "      <td>ERR1726523</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-14.070426</td>\n",
       "      <td>3.470203</td>\n",
       "      <td>0.075315</td>\n",
       "      <td>0.032361</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>12.005218</td>\n",
       "      <td>-114.248528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>28</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2621794</td>\n",
       "      <td>ERS491436</td>\n",
       "      <td>ERX1796959</td>\n",
       "      <td>ERR1726524</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>4.738948</td>\n",
       "      <td>-4.678504</td>\n",
       "      <td>0.061918</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.250000</td>\n",
       "      <td>-72.642651</td>\n",
       "      <td>-35.181114</td>\n",
       "      <td>63.0</td>\n",
       "      <td>-34.302478</td>\n",
       "      <td>-73.496846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>29</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2732548</td>\n",
       "      <td>ERS492060</td>\n",
       "      <td>ERX1796611</td>\n",
       "      <td>ERR1726525</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>7.105469</td>\n",
       "      <td>3.096856</td>\n",
       "      <td>-0.041450</td>\n",
       "      <td>0.138825</td>\n",
       "      <td>4.166667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>4.631580</td>\n",
       "      <td>-82.454814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>792</th>\n",
       "      <td>793</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623667</td>\n",
       "      <td>ERS494226</td>\n",
       "      <td>ERX948013</td>\n",
       "      <td>ERR868495</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>1.373768</td>\n",
       "      <td>4.126204</td>\n",
       "      <td>-0.044917</td>\n",
       "      <td>0.144593</td>\n",
       "      <td>3.472222</td>\n",
       "      <td>-70.706411</td>\n",
       "      <td>39.957414</td>\n",
       "      <td>24.0</td>\n",
       "      <td>39.495355</td>\n",
       "      <td>-69.744626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>793</th>\n",
       "      <td>794</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619754</td>\n",
       "      <td>ERS488460</td>\n",
       "      <td>ERX948027</td>\n",
       "      <td>ERR868496</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>3.680158</td>\n",
       "      <td>16.845059</td>\n",
       "      <td>0.054261</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>794</th>\n",
       "      <td>795</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619443</td>\n",
       "      <td>ERS487982</td>\n",
       "      <td>ERX947980</td>\n",
       "      <td>ERR868497</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>66.915191</td>\n",
       "      <td>24.270961</td>\n",
       "      <td>-10.417380</td>\n",
       "      <td>0.284530</td>\n",
       "      <td>13.638889</td>\n",
       "      <td>-4.201844</td>\n",
       "      <td>35.489171</td>\n",
       "      <td>42.0</td>\n",
       "      <td>35.673706</td>\n",
       "      <td>-4.145546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>795</th>\n",
       "      <td>796</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620010</td>\n",
       "      <td>ERS488809</td>\n",
       "      <td>ERX948026</td>\n",
       "      <td>ERR868498</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>10.755607</td>\n",
       "      <td>-6.652039</td>\n",
       "      <td>0.005785</td>\n",
       "      <td>0.069484</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>20.258041</td>\n",
       "      <td>64.562537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>796</th>\n",
       "      <td>797</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622010</td>\n",
       "      <td>ERS491719</td>\n",
       "      <td>ERX948101</td>\n",
       "      <td>ERR868499</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>14.212952</td>\n",
       "      <td>1.275138</td>\n",
       "      <td>-0.006839</td>\n",
       "      <td>0.028784</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-25.231969</td>\n",
       "      <td>-112.818256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>797</th>\n",
       "      <td>798</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622997</td>\n",
       "      <td>ERS493199</td>\n",
       "      <td>ERX948090</td>\n",
       "      <td>ERR868500</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.155494</td>\n",
       "      <td>-6.599451</td>\n",
       "      <td>-0.181183</td>\n",
       "      <td>0.100184</td>\n",
       "      <td>1.611111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>11.448054</td>\n",
       "      <td>-150.695800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>798</th>\n",
       "      <td>799</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622536</td>\n",
       "      <td>ERS492426</td>\n",
       "      <td>ERX947963</td>\n",
       "      <td>ERR868501</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-5.293649</td>\n",
       "      <td>-4.682669</td>\n",
       "      <td>-0.120564</td>\n",
       "      <td>0.040113</td>\n",
       "      <td>11.083333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-23.568914</td>\n",
       "      <td>-127.310923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>799</th>\n",
       "      <td>800</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622258</td>\n",
       "      <td>ERS492052</td>\n",
       "      <td>ERX947951</td>\n",
       "      <td>ERR868502</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>7.105469</td>\n",
       "      <td>3.096856</td>\n",
       "      <td>-0.041450</td>\n",
       "      <td>0.138825</td>\n",
       "      <td>4.166667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>4.631580</td>\n",
       "      <td>-82.454814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>800</th>\n",
       "      <td>801</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620010</td>\n",
       "      <td>ERS488809</td>\n",
       "      <td>ERX948057</td>\n",
       "      <td>ERR868503</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>10.755607</td>\n",
       "      <td>-6.652039</td>\n",
       "      <td>0.005785</td>\n",
       "      <td>0.069484</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>20.258041</td>\n",
       "      <td>64.562537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>801</th>\n",
       "      <td>802</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619723</td>\n",
       "      <td>ERS488410</td>\n",
       "      <td>ERX948015</td>\n",
       "      <td>ERR868504</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-5.127640</td>\n",
       "      <td>-0.375367</td>\n",
       "      <td>0.051438</td>\n",
       "      <td>0.087535</td>\n",
       "      <td>1.333333</td>\n",
       "      <td>14.973150</td>\n",
       "      <td>34.451200</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32.144171</td>\n",
       "      <td>16.847326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>802</th>\n",
       "      <td>803</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622278</td>\n",
       "      <td>ERS492090</td>\n",
       "      <td>ERX948003</td>\n",
       "      <td>ERR868505</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>7.105469</td>\n",
       "      <td>3.096856</td>\n",
       "      <td>-0.041450</td>\n",
       "      <td>0.138825</td>\n",
       "      <td>4.166667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>4.631580</td>\n",
       "      <td>-82.454814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>803</th>\n",
       "      <td>804</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622369</td>\n",
       "      <td>ERS492212</td>\n",
       "      <td>ERX948066</td>\n",
       "      <td>ERR868506</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-44.758723</td>\n",
       "      <td>-0.243588</td>\n",
       "      <td>-0.002594</td>\n",
       "      <td>0.005026</td>\n",
       "      <td>2.083333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>6.165688</td>\n",
       "      <td>-92.915797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>804</th>\n",
       "      <td>805</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620354</td>\n",
       "      <td>ERS489251</td>\n",
       "      <td>ERX948006</td>\n",
       "      <td>ERR868507</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>20.311192</td>\n",
       "      <td>-18.737291</td>\n",
       "      <td>0.111952</td>\n",
       "      <td>0.069572</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>72.715398</td>\n",
       "      <td>3.753862</td>\n",
       "      <td>31.0</td>\n",
       "      <td>2.260611</td>\n",
       "      <td>75.796797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>805</th>\n",
       "      <td>806</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2620292</td>\n",
       "      <td>ERS489175</td>\n",
       "      <td>ERX947995</td>\n",
       "      <td>ERR868508</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-38.085507</td>\n",
       "      <td>-6.593482</td>\n",
       "      <td>0.146439</td>\n",
       "      <td>0.118141</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>73.488950</td>\n",
       "      <td>4.660350</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>806</th>\n",
       "      <td>807</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623564</td>\n",
       "      <td>ERS494087</td>\n",
       "      <td>ERX948073</td>\n",
       "      <td>ERR868509</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>8.319742</td>\n",
       "      <td>91.787857</td>\n",
       "      <td>-4.707638</td>\n",
       "      <td>0.398799</td>\n",
       "      <td>1.083333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>808</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619419</td>\n",
       "      <td>ERS487956</td>\n",
       "      <td>ERX947959</td>\n",
       "      <td>ERR868510</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.031746</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>-6.552800</td>\n",
       "      <td>36.563150</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>809</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623839</td>\n",
       "      <td>ERS494501</td>\n",
       "      <td>ERX948078</td>\n",
       "      <td>ERR868511</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>11.224682</td>\n",
       "      <td>4.559712</td>\n",
       "      <td>-0.032539</td>\n",
       "      <td>0.073051</td>\n",
       "      <td>1.388889</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>36.403496</td>\n",
       "      <td>-38.208687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>809</th>\n",
       "      <td>810</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623321</td>\n",
       "      <td>ERS493712</td>\n",
       "      <td>ERX948018</td>\n",
       "      <td>ERR868512</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-14.070426</td>\n",
       "      <td>3.470203</td>\n",
       "      <td>0.075315</td>\n",
       "      <td>0.032361</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>12.005218</td>\n",
       "      <td>-114.248528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>811</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622661</td>\n",
       "      <td>ERS492651</td>\n",
       "      <td>ERX948010</td>\n",
       "      <td>ERR868513</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-13.524453</td>\n",
       "      <td>5.403076</td>\n",
       "      <td>0.007751</td>\n",
       "      <td>0.028977</td>\n",
       "      <td>15.833333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-11.599533</td>\n",
       "      <td>-137.896881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>812</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622679</td>\n",
       "      <td>ERS492686</td>\n",
       "      <td>ERX953564</td>\n",
       "      <td>ERR873959</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-13.524453</td>\n",
       "      <td>5.403076</td>\n",
       "      <td>0.007751</td>\n",
       "      <td>0.028977</td>\n",
       "      <td>15.833333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-11.599533</td>\n",
       "      <td>-137.896881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>812</th>\n",
       "      <td>813</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623246</td>\n",
       "      <td>ERS493592</td>\n",
       "      <td>ERX953568</td>\n",
       "      <td>ERR873960</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-20.748995</td>\n",
       "      <td>1.721794</td>\n",
       "      <td>-1.155182</td>\n",
       "      <td>0.047316</td>\n",
       "      <td>41.666667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>33.472302</td>\n",
       "      <td>-120.652510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>813</th>\n",
       "      <td>814</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622038</td>\n",
       "      <td>ERS491757</td>\n",
       "      <td>ERX953562</td>\n",
       "      <td>ERR873961</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>14.212952</td>\n",
       "      <td>1.275138</td>\n",
       "      <td>-0.006839</td>\n",
       "      <td>0.028784</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-25.231969</td>\n",
       "      <td>-112.818256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>814</th>\n",
       "      <td>815</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619981</td>\n",
       "      <td>ERS488780</td>\n",
       "      <td>ERX953565</td>\n",
       "      <td>ERR873962</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>2.996854</td>\n",
       "      <td>8.555371</td>\n",
       "      <td>-0.087532</td>\n",
       "      <td>0.061870</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>61.900938</td>\n",
       "      <td>24.819492</td>\n",
       "      <td>50.0</td>\n",
       "      <td>24.428166</td>\n",
       "      <td>61.449162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>815</th>\n",
       "      <td>816</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622511</td>\n",
       "      <td>ERS492393</td>\n",
       "      <td>ERX953572</td>\n",
       "      <td>ERR873963</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.395135</td>\n",
       "      <td>6.423670</td>\n",
       "      <td>-0.075767</td>\n",
       "      <td>0.018781</td>\n",
       "      <td>34.305556</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-16.755721</td>\n",
       "      <td>-99.729821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>816</th>\n",
       "      <td>817</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623603</td>\n",
       "      <td>ERS494131</td>\n",
       "      <td>ERX953566</td>\n",
       "      <td>ERR873964</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>47.329031</td>\n",
       "      <td>34.457153</td>\n",
       "      <td>2.028463</td>\n",
       "      <td>0.249201</td>\n",
       "      <td>1.861111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>29.041205</td>\n",
       "      <td>-79.321465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>817</th>\n",
       "      <td>818</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623579</td>\n",
       "      <td>ERS494105</td>\n",
       "      <td>ERX953567</td>\n",
       "      <td>ERR873965</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>8.319742</td>\n",
       "      <td>91.787857</td>\n",
       "      <td>-4.707638</td>\n",
       "      <td>0.398799</td>\n",
       "      <td>1.083333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>818</th>\n",
       "      <td>819</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622038</td>\n",
       "      <td>ERS491757</td>\n",
       "      <td>ERX953571</td>\n",
       "      <td>ERR873966</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>14.212952</td>\n",
       "      <td>1.275138</td>\n",
       "      <td>-0.006839</td>\n",
       "      <td>0.028784</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-25.231969</td>\n",
       "      <td>-112.818256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>819</th>\n",
       "      <td>820</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2623246</td>\n",
       "      <td>ERS493592</td>\n",
       "      <td>ERX953569</td>\n",
       "      <td>ERR873967</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-20.748995</td>\n",
       "      <td>1.721794</td>\n",
       "      <td>-1.155182</td>\n",
       "      <td>0.047316</td>\n",
       "      <td>41.666667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>33.472302</td>\n",
       "      <td>-120.652510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>820</th>\n",
       "      <td>821</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2622679</td>\n",
       "      <td>ERS492686</td>\n",
       "      <td>ERX953570</td>\n",
       "      <td>ERR873968</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>-13.524453</td>\n",
       "      <td>5.403076</td>\n",
       "      <td>0.007751</td>\n",
       "      <td>0.028977</td>\n",
       "      <td>15.833333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>120.0</td>\n",
       "      <td>-11.599533</td>\n",
       "      <td>-137.896881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>821</th>\n",
       "      <td>822</td>\n",
       "      <td>PRJEB4352</td>\n",
       "      <td>SAMEA2619981</td>\n",
       "      <td>ERS488780</td>\n",
       "      <td>ERX953563</td>\n",
       "      <td>ERR873969</td>\n",
       "      <td>408172</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>...</td>\n",
       "      <td>2.996854</td>\n",
       "      <td>8.555371</td>\n",
       "      <td>-0.087532</td>\n",
       "      <td>0.061870</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>61.900938</td>\n",
       "      <td>24.819492</td>\n",
       "      <td>50.0</td>\n",
       "      <td>24.428166</td>\n",
       "      <td>61.449162</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>822 rows × 96 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0 study_accession sample_accession secondary_sample_accession  \\\n",
       "0             0       PRJEB4352     SAMEA2656965                  ERS488900   \n",
       "1             1       PRJEB4352     SAMEA2620205                  ERS489054   \n",
       "2             2       PRJEB4352     SAMEA2623506                  ERS493999   \n",
       "3             3       PRJEB4352     SAMEA2620172                  ERS489013   \n",
       "4             4       PRJEB4352     SAMEA2730950                  ERS494049   \n",
       "5             5       PRJEB4352     SAMEA2619554                  ERS488153   \n",
       "6             6       PRJEB4352     SAMEA2619623                  ERS488251   \n",
       "7             7       PRJEB4352     SAMEA2619725                  ERS488412   \n",
       "8             8       PRJEB4352     SAMEA2622039                  ERS491758   \n",
       "9             9       PRJEB4352     SAMEA2621474                  ERS490981   \n",
       "10           10       PRJEB4352     SAMEA2620072                  ERS488886   \n",
       "11           11       PRJEB4352     SAMEA2619421                  ERS487958   \n",
       "12           12       PRJEB4352     SAMEA2621827                  ERS491478   \n",
       "13           13       PRJEB4352     SAMEA2731011                  ERS494136   \n",
       "14           14       PRJEB4352     SAMEA2622132                  ERS491887   \n",
       "15           15       PRJEB4352     SAMEA2619959                  ERS488754   \n",
       "16           16       PRJEB4352     SAMEA2622039                  ERS491758   \n",
       "17           17       PRJEB4352     SAMEA2657022                  ERS488820   \n",
       "18           18       PRJEB4352     SAMEA2657022                  ERS488820   \n",
       "19           19       PRJEB4352     SAMEA2619421                  ERS487958   \n",
       "20           20       PRJEB4352     SAMEA2657050                  ERS489336   \n",
       "21           21       PRJEB4352     SAMEA2732453                  ERS493776   \n",
       "22           22       PRJEB4352     SAMEA2657097                  ERS492000   \n",
       "23           23       PRJEB4352     SAMEA2620222                  ERS489079   \n",
       "24           24       PRJEB4352     SAMEA2619789                  ERS488516   \n",
       "25           25       PRJEB4352     SAMEA2656981                  ERS488816   \n",
       "26           26       PRJEB4352     SAMEA2657109                  ERS492751   \n",
       "27           27       PRJEB4352     SAMEA2623302                  ERS493677   \n",
       "28           28       PRJEB4352     SAMEA2621794                  ERS491436   \n",
       "29           29       PRJEB4352     SAMEA2732548                  ERS492060   \n",
       "..          ...             ...              ...                        ...   \n",
       "792         793       PRJEB4352     SAMEA2623667                  ERS494226   \n",
       "793         794       PRJEB4352     SAMEA2619754                  ERS488460   \n",
       "794         795       PRJEB4352     SAMEA2619443                  ERS487982   \n",
       "795         796       PRJEB4352     SAMEA2620010                  ERS488809   \n",
       "796         797       PRJEB4352     SAMEA2622010                  ERS491719   \n",
       "797         798       PRJEB4352     SAMEA2622997                  ERS493199   \n",
       "798         799       PRJEB4352     SAMEA2622536                  ERS492426   \n",
       "799         800       PRJEB4352     SAMEA2622258                  ERS492052   \n",
       "800         801       PRJEB4352     SAMEA2620010                  ERS488809   \n",
       "801         802       PRJEB4352     SAMEA2619723                  ERS488410   \n",
       "802         803       PRJEB4352     SAMEA2622278                  ERS492090   \n",
       "803         804       PRJEB4352     SAMEA2622369                  ERS492212   \n",
       "804         805       PRJEB4352     SAMEA2620354                  ERS489251   \n",
       "805         806       PRJEB4352     SAMEA2620292                  ERS489175   \n",
       "806         807       PRJEB4352     SAMEA2623564                  ERS494087   \n",
       "807         808       PRJEB4352     SAMEA2619419                  ERS487956   \n",
       "808         809       PRJEB4352     SAMEA2623839                  ERS494501   \n",
       "809         810       PRJEB4352     SAMEA2623321                  ERS493712   \n",
       "810         811       PRJEB4352     SAMEA2622661                  ERS492651   \n",
       "811         812       PRJEB4352     SAMEA2622679                  ERS492686   \n",
       "812         813       PRJEB4352     SAMEA2623246                  ERS493592   \n",
       "813         814       PRJEB4352     SAMEA2622038                  ERS491757   \n",
       "814         815       PRJEB4352     SAMEA2619981                  ERS488780   \n",
       "815         816       PRJEB4352     SAMEA2622511                  ERS492393   \n",
       "816         817       PRJEB4352     SAMEA2623603                  ERS494131   \n",
       "817         818       PRJEB4352     SAMEA2623579                  ERS494105   \n",
       "818         819       PRJEB4352     SAMEA2622038                  ERS491757   \n",
       "819         820       PRJEB4352     SAMEA2623246                  ERS493592   \n",
       "820         821       PRJEB4352     SAMEA2622679                  ERS492686   \n",
       "821         822       PRJEB4352     SAMEA2619981                  ERS488780   \n",
       "\n",
       "    experiment_accession run_accession  tax_id    scientific_name  \\\n",
       "0             ERX1771177    ERR1700889  408172  marine metagenome   \n",
       "1             ERX1771189    ERR1700890  408172  marine metagenome   \n",
       "2             ERX1771187    ERR1700891  408172  marine metagenome   \n",
       "3             ERX1771191    ERR1700892  408172  marine metagenome   \n",
       "4             ERX1771202    ERR1700893  408172  marine metagenome   \n",
       "5             ERX1771192    ERR1700894  408172  marine metagenome   \n",
       "6             ERX1771178    ERR1700895  408172  marine metagenome   \n",
       "7             ERX1771195    ERR1700896  408172  marine metagenome   \n",
       "8             ERX1771199    ERR1700897  408172  marine metagenome   \n",
       "9             ERX1771201    ERR1700898  408172  marine metagenome   \n",
       "10            ERX1771182    ERR1700899  408172  marine metagenome   \n",
       "11            ERX1771180    ERR1700900  408172  marine metagenome   \n",
       "12            ERX1771186    ERR1700901  408172  marine metagenome   \n",
       "13            ERX1771181    ERR1700902  408172  marine metagenome   \n",
       "14            ERX1771196    ERR1700903  408172  marine metagenome   \n",
       "15            ERX1771197    ERR1700904  408172  marine metagenome   \n",
       "16            ERX1771185    ERR1700905  408172  marine metagenome   \n",
       "17            ERX1771190    ERR1700906  408172  marine metagenome   \n",
       "18            ERX1771188    ERR1700907  408172  marine metagenome   \n",
       "19            ERX1771198    ERR1700908  408172  marine metagenome   \n",
       "20            ERX1771193    ERR1700909  408172  marine metagenome   \n",
       "21            ERX1771200    ERR1700910  408172  marine metagenome   \n",
       "22            ERX1771184    ERR1700911  408172  marine metagenome   \n",
       "23            ERX1771194    ERR1700912  408172  marine metagenome   \n",
       "24            ERX1771183    ERR1700913  408172  marine metagenome   \n",
       "25            ERX1771179    ERR1700914  408172  marine metagenome   \n",
       "26            ERX1796605    ERR1726522  408172  marine metagenome   \n",
       "27            ERX1796551    ERR1726523  408172  marine metagenome   \n",
       "28            ERX1796959    ERR1726524  408172  marine metagenome   \n",
       "29            ERX1796611    ERR1726525  408172  marine metagenome   \n",
       "..                   ...           ...     ...                ...   \n",
       "792            ERX948013     ERR868495  408172  marine metagenome   \n",
       "793            ERX948027     ERR868496  408172  marine metagenome   \n",
       "794            ERX947980     ERR868497  408172  marine metagenome   \n",
       "795            ERX948026     ERR868498  408172  marine metagenome   \n",
       "796            ERX948101     ERR868499  408172  marine metagenome   \n",
       "797            ERX948090     ERR868500  408172  marine metagenome   \n",
       "798            ERX947963     ERR868501  408172  marine metagenome   \n",
       "799            ERX947951     ERR868502  408172  marine metagenome   \n",
       "800            ERX948057     ERR868503  408172  marine metagenome   \n",
       "801            ERX948015     ERR868504  408172  marine metagenome   \n",
       "802            ERX948003     ERR868505  408172  marine metagenome   \n",
       "803            ERX948066     ERR868506  408172  marine metagenome   \n",
       "804            ERX948006     ERR868507  408172  marine metagenome   \n",
       "805            ERX947995     ERR868508  408172  marine metagenome   \n",
       "806            ERX948073     ERR868509  408172  marine metagenome   \n",
       "807            ERX947959     ERR868510  408172  marine metagenome   \n",
       "808            ERX948078     ERR868511  408172  marine metagenome   \n",
       "809            ERX948018     ERR868512  408172  marine metagenome   \n",
       "810            ERX948010     ERR868513  408172  marine metagenome   \n",
       "811            ERX953564     ERR873959  408172  marine metagenome   \n",
       "812            ERX953568     ERR873960  408172  marine metagenome   \n",
       "813            ERX953562     ERR873961  408172  marine metagenome   \n",
       "814            ERX953565     ERR873962  408172  marine metagenome   \n",
       "815            ERX953572     ERR873963  408172  marine metagenome   \n",
       "816            ERX953566     ERR873964  408172  marine metagenome   \n",
       "817            ERX953567     ERR873965  408172  marine metagenome   \n",
       "818            ERX953571     ERR873966  408172  marine metagenome   \n",
       "819            ERX953569     ERR873967  408172  marine metagenome   \n",
       "820            ERX953570     ERR873968  408172  marine metagenome   \n",
       "821            ERX953563     ERR873969  408172  marine metagenome   \n",
       "\n",
       "        instrument_model library_layout  ...   u [cm/s]   v [cm/s]         OW  \\\n",
       "0    Illumina HiSeq 2000         PAIRED  ...  -0.557961  15.046586   0.403157   \n",
       "1    Illumina HiSeq 2000         PAIRED  ...  -0.466550   4.306080  -0.412443   \n",
       "2    Illumina HiSeq 2000         PAIRED  ...  28.262368   8.317923  -2.922481   \n",
       "3    Illumina HiSeq 2000         PAIRED  ...  33.111660 -33.418508   0.043242   \n",
       "4    Illumina HiSeq 2000         PAIRED  ...   8.319742  91.787857  -4.707638   \n",
       "5    Illumina HiSeq 2000         PAIRED  ... -16.430919   4.368810   0.021447   \n",
       "6    Illumina HiSeq 2000         PAIRED  ...        NaN        NaN        NaN   \n",
       "7    Illumina HiSeq 2000         PAIRED  ...  -5.127640  -0.375367   0.051438   \n",
       "8    Illumina HiSeq 2000         PAIRED  ...  14.212952   1.275138  -0.006839   \n",
       "9    Illumina HiSeq 2000         PAIRED  ...  10.611410   1.423920   0.072135   \n",
       "10   Illumina HiSeq 2000         PAIRED  ...  -0.557961  15.046586   0.403157   \n",
       "11   Illumina HiSeq 2000         PAIRED  ...        NaN        NaN        NaN   \n",
       "12   Illumina HiSeq 2000         PAIRED  ...   4.738948  -4.678504   0.061918   \n",
       "13   Illumina HiSeq 2000         PAIRED  ...  47.329031  34.457153   2.028463   \n",
       "14   Illumina HiSeq 2000         PAIRED  ...  -0.841770   2.208339   0.013354   \n",
       "15   Illumina HiSeq 2000         PAIRED  ...   3.924148   8.325847  -0.030610   \n",
       "16   Illumina HiSeq 2000         PAIRED  ...  14.212952   1.275138  -0.006839   \n",
       "17   Illumina HiSeq 2000         PAIRED  ...  10.755607  -6.652039   0.005785   \n",
       "18   Illumina HiSeq 2000         PAIRED  ...  10.755607  -6.652039   0.005785   \n",
       "19   Illumina HiSeq 2000         PAIRED  ...        NaN        NaN        NaN   \n",
       "20   Illumina HiSeq 2000         PAIRED  ...   3.682729  19.395807  -0.118280   \n",
       "21   Illumina HiSeq 2000         PAIRED  ...  52.020196   1.040314  -0.002795   \n",
       "22   Illumina HiSeq 2000         PAIRED  ... -12.746951  15.658745   0.043188   \n",
       "23   Illumina HiSeq 2000         PAIRED  ...  -0.466550   4.306080  -0.412443   \n",
       "24   Illumina HiSeq 2000         PAIRED  ...  -9.449987   5.586469   0.369749   \n",
       "25   Illumina HiSeq 2000         PAIRED  ...  10.755607  -6.652039   0.005785   \n",
       "26   Illumina HiSeq 2000         PAIRED  ...  -4.286042   2.890793  -0.011503   \n",
       "27   Illumina HiSeq 2000         PAIRED  ... -14.070426   3.470203   0.075315   \n",
       "28   Illumina HiSeq 2000         PAIRED  ...   4.738948  -4.678504   0.061918   \n",
       "29   Illumina HiSeq 2000         PAIRED  ...   7.105469   3.096856  -0.041450   \n",
       "..                   ...            ...  ...        ...        ...        ...   \n",
       "792  Illumina HiSeq 2000         PAIRED  ...   1.373768   4.126204  -0.044917   \n",
       "793  Illumina HiSeq 2000         PAIRED  ...   3.680158  16.845059   0.054261   \n",
       "794  Illumina HiSeq 2000         PAIRED  ...  66.915191  24.270961 -10.417380   \n",
       "795  Illumina HiSeq 2000         PAIRED  ...  10.755607  -6.652039   0.005785   \n",
       "796  Illumina HiSeq 2000         PAIRED  ...  14.212952   1.275138  -0.006839   \n",
       "797  Illumina HiSeq 2000         PAIRED  ...  -1.155494  -6.599451  -0.181183   \n",
       "798  Illumina HiSeq 2000         PAIRED  ...  -5.293649  -4.682669  -0.120564   \n",
       "799  Illumina HiSeq 2000         PAIRED  ...   7.105469   3.096856  -0.041450   \n",
       "800  Illumina HiSeq 2000         PAIRED  ...  10.755607  -6.652039   0.005785   \n",
       "801  Illumina HiSeq 2000         PAIRED  ...  -5.127640  -0.375367   0.051438   \n",
       "802  Illumina HiSeq 2000         PAIRED  ...   7.105469   3.096856  -0.041450   \n",
       "803  Illumina HiSeq 2000         PAIRED  ... -44.758723  -0.243588  -0.002594   \n",
       "804  Illumina HiSeq 2000         PAIRED  ...  20.311192 -18.737291   0.111952   \n",
       "805  Illumina HiSeq 2000         PAIRED  ... -38.085507  -6.593482   0.146439   \n",
       "806  Illumina HiSeq 2000         PAIRED  ...   8.319742  91.787857  -4.707638   \n",
       "807  Illumina HiSeq 2000         PAIRED  ...        NaN        NaN        NaN   \n",
       "808  Illumina HiSeq 2000         PAIRED  ...  11.224682   4.559712  -0.032539   \n",
       "809  Illumina HiSeq 2000         PAIRED  ... -14.070426   3.470203   0.075315   \n",
       "810  Illumina HiSeq 2000         PAIRED  ... -13.524453   5.403076   0.007751   \n",
       "811  Illumina HiSeq 2000         PAIRED  ... -13.524453   5.403076   0.007751   \n",
       "812  Illumina HiSeq 2000         PAIRED  ... -20.748995   1.721794  -1.155182   \n",
       "813  Illumina HiSeq 2000         PAIRED  ...  14.212952   1.275138  -0.006839   \n",
       "814  Illumina HiSeq 2000         PAIRED  ...   2.996854   8.555371  -0.087532   \n",
       "815  Illumina HiSeq 2000         PAIRED  ...  -2.395135   6.423670  -0.075767   \n",
       "816  Illumina HiSeq 2000         PAIRED  ...  47.329031  34.457153   2.028463   \n",
       "817  Illumina HiSeq 2000         PAIRED  ...   8.319742  91.787857  -4.707638   \n",
       "818  Illumina HiSeq 2000         PAIRED  ...  14.212952   1.275138  -0.006839   \n",
       "819  Illumina HiSeq 2000         PAIRED  ... -20.748995   1.721794  -1.155182   \n",
       "820  Illumina HiSeq 2000         PAIRED  ... -13.524453   5.403076   0.007751   \n",
       "821  Illumina HiSeq 2000         PAIRED  ...   2.996854   8.555371  -0.087532   \n",
       "\n",
       "    MLE [1/day]  RT [days] Latitude (of the continental shelf with...)  \\\n",
       "0           NaN   2.250000                                         NaN   \n",
       "1      0.026178  19.027778                                         NaN   \n",
       "2      0.044572  25.750000                                         NaN   \n",
       "3      0.086921   0.000000                                         NaN   \n",
       "4      0.398799   1.083333                                         NaN   \n",
       "5      0.190651   0.000000                                         NaN   \n",
       "6      0.000000   0.000000                                    2.797800   \n",
       "7      0.087535   1.333333                                   14.973150   \n",
       "8      0.028784   0.000000                                         NaN   \n",
       "9      0.000000   0.000000                                  -65.022700   \n",
       "10          NaN   2.250000                                         NaN   \n",
       "11     0.031746   4.000000                                   -6.552800   \n",
       "12          NaN   3.250000                                  -72.642651   \n",
       "13     0.249201   1.861111                                         NaN   \n",
       "14     0.045278   6.000000                                         NaN   \n",
       "15     0.074837   0.000000                                         NaN   \n",
       "16     0.028784   0.000000                                         NaN   \n",
       "17     0.069484   0.000000                                         NaN   \n",
       "18     0.069484   0.000000                                         NaN   \n",
       "19     0.031746   4.000000                                   -6.552800   \n",
       "20     0.067544   3.777778                                         NaN   \n",
       "21     0.020154   0.000000                                         NaN   \n",
       "22     0.012324   0.000000                                         NaN   \n",
       "23     0.026178  19.027778                                         NaN   \n",
       "24     0.000000   0.000000                                         NaN   \n",
       "25     0.069484   0.000000                                         NaN   \n",
       "26     0.053774   1.083333                                         NaN   \n",
       "27     0.032361   0.000000                                         NaN   \n",
       "28          NaN   3.250000                                  -72.642651   \n",
       "29     0.138825   4.166667                                         NaN   \n",
       "..          ...        ...                                         ...   \n",
       "792    0.144593   3.472222                                  -70.706411   \n",
       "793    0.000000   0.000000                                         NaN   \n",
       "794    0.284530  13.638889                                   -4.201844   \n",
       "795    0.069484   0.000000                                         NaN   \n",
       "796    0.028784   0.000000                                         NaN   \n",
       "797    0.100184   1.611111                                         NaN   \n",
       "798    0.040113  11.083333                                         NaN   \n",
       "799    0.138825   4.166667                                         NaN   \n",
       "800    0.069484   0.000000                                         NaN   \n",
       "801    0.087535   1.333333                                   14.973150   \n",
       "802    0.138825   4.166667                                         NaN   \n",
       "803    0.005026   2.083333                                         NaN   \n",
       "804    0.069572   0.000000                                   72.715398   \n",
       "805    0.118141   2.000000                                   73.488950   \n",
       "806    0.398799   1.083333                                         NaN   \n",
       "807    0.031746   4.000000                                   -6.552800   \n",
       "808    0.073051   1.388889                                         NaN   \n",
       "809    0.032361   0.000000                                         NaN   \n",
       "810    0.028977  15.833333                                         NaN   \n",
       "811    0.028977  15.833333                                         NaN   \n",
       "812    0.047316  41.666667                                         NaN   \n",
       "813    0.028784   0.000000                                         NaN   \n",
       "814    0.061870   0.000000                                   61.900938   \n",
       "815    0.018781  34.305556                                         NaN   \n",
       "816    0.249201   1.861111                                         NaN   \n",
       "817    0.398799   1.083333                                         NaN   \n",
       "818    0.028784   0.000000                                         NaN   \n",
       "819    0.047316  41.666667                                         NaN   \n",
       "820    0.028977  15.833333                                         NaN   \n",
       "821    0.061870   0.000000                                   61.900938   \n",
       "\n",
       "    Longitude (of the continental shelf with...) Age [days]  \\\n",
       "0                                            NaN      120.0   \n",
       "1                                            NaN      120.0   \n",
       "2                                            NaN      120.0   \n",
       "3                                            NaN      120.0   \n",
       "4                                            NaN      120.0   \n",
       "5                                            NaN      120.0   \n",
       "6                                      41.665900        0.0   \n",
       "7                                      34.451200        0.0   \n",
       "8                                            NaN      120.0   \n",
       "9                                     -54.418000        0.0   \n",
       "10                                           NaN      120.0   \n",
       "11                                     36.563150        0.0   \n",
       "12                                    -35.181114       63.0   \n",
       "13                                           NaN      120.0   \n",
       "14                                           NaN      120.0   \n",
       "15                                           NaN      120.0   \n",
       "16                                           NaN      120.0   \n",
       "17                                           NaN      120.0   \n",
       "18                                           NaN      120.0   \n",
       "19                                     36.563150        0.0   \n",
       "20                                           NaN      120.0   \n",
       "21                                           NaN      120.0   \n",
       "22                                           NaN      120.0   \n",
       "23                                           NaN      120.0   \n",
       "24                                           NaN      120.0   \n",
       "25                                           NaN      120.0   \n",
       "26                                           NaN      120.0   \n",
       "27                                           NaN      120.0   \n",
       "28                                    -35.181114       63.0   \n",
       "29                                           NaN      120.0   \n",
       "..                                           ...        ...   \n",
       "792                                    39.957414       24.0   \n",
       "793                                          NaN      120.0   \n",
       "794                                    35.489171       42.0   \n",
       "795                                          NaN      120.0   \n",
       "796                                          NaN      120.0   \n",
       "797                                          NaN      120.0   \n",
       "798                                          NaN      120.0   \n",
       "799                                          NaN      120.0   \n",
       "800                                          NaN      120.0   \n",
       "801                                    34.451200        0.0   \n",
       "802                                          NaN      120.0   \n",
       "803                                          NaN      120.0   \n",
       "804                                     3.753862       31.0   \n",
       "805                                     4.660350        0.0   \n",
       "806                                          NaN      120.0   \n",
       "807                                    36.563150        0.0   \n",
       "808                                          NaN      120.0   \n",
       "809                                          NaN      120.0   \n",
       "810                                          NaN      120.0   \n",
       "811                                          NaN      120.0   \n",
       "812                                          NaN      120.0   \n",
       "813                                          NaN      120.0   \n",
       "814                                    24.819492       50.0   \n",
       "815                                          NaN      120.0   \n",
       "816                                          NaN      120.0   \n",
       "817                                          NaN      120.0   \n",
       "818                                          NaN      120.0   \n",
       "819                                          NaN      120.0   \n",
       "820                                          NaN      120.0   \n",
       "821                                    24.819492       50.0   \n",
       "\n",
       "    Latitude (of the geographic coordinate ...)  \\\n",
       "0                                     16.583327   \n",
       "1                                     14.404658   \n",
       "2                                     26.338683   \n",
       "3                                     17.386625   \n",
       "4                                           NaN   \n",
       "5                                     38.821749   \n",
       "6                                           NaN   \n",
       "7                                     32.144171   \n",
       "8                                    -25.231969   \n",
       "9                                           NaN   \n",
       "10                                    16.583327   \n",
       "11                                          NaN   \n",
       "12                                   -34.302478   \n",
       "13                                    29.041205   \n",
       "14                                   -13.259274   \n",
       "15                                    23.584126   \n",
       "16                                   -25.231969   \n",
       "17                                    20.258041   \n",
       "18                                    20.258041   \n",
       "19                                          NaN   \n",
       "20                                   -14.979369   \n",
       "21                                     7.512087   \n",
       "22                                    -9.444510   \n",
       "23                                    14.404658   \n",
       "24                                          NaN   \n",
       "25                                    20.258041   \n",
       "26                                    -8.878592   \n",
       "27                                    12.005218   \n",
       "28                                   -34.302478   \n",
       "29                                     4.631580   \n",
       "..                                          ...   \n",
       "792                                   39.495355   \n",
       "793                                         NaN   \n",
       "794                                   35.673706   \n",
       "795                                   20.258041   \n",
       "796                                  -25.231969   \n",
       "797                                   11.448054   \n",
       "798                                  -23.568914   \n",
       "799                                    4.631580   \n",
       "800                                   20.258041   \n",
       "801                                   32.144171   \n",
       "802                                    4.631580   \n",
       "803                                    6.165688   \n",
       "804                                    2.260611   \n",
       "805                                         NaN   \n",
       "806                                         NaN   \n",
       "807                                         NaN   \n",
       "808                                   36.403496   \n",
       "809                                   12.005218   \n",
       "810                                  -11.599533   \n",
       "811                                  -11.599533   \n",
       "812                                   33.472302   \n",
       "813                                  -25.231969   \n",
       "814                                   24.428166   \n",
       "815                                  -16.755721   \n",
       "816                                   29.041205   \n",
       "817                                         NaN   \n",
       "818                                  -25.231969   \n",
       "819                                   33.472302   \n",
       "820                                  -11.599533   \n",
       "821                                   24.428166   \n",
       "\n",
       "    Longitude (of the geographic coordinate ...)  \n",
       "0                                      63.625415  \n",
       "1                                      69.689377  \n",
       "2                                     -87.272865  \n",
       "3                                      66.210146  \n",
       "4                                            NaN  \n",
       "5                                       6.902024  \n",
       "6                                            NaN  \n",
       "7                                      16.847326  \n",
       "8                                    -112.818256  \n",
       "9                                            NaN  \n",
       "10                                     63.625415  \n",
       "11                                           NaN  \n",
       "12                                    -73.496846  \n",
       "13                                    -79.321465  \n",
       "14                                    -94.231260  \n",
       "15                                     60.241719  \n",
       "16                                   -112.818256  \n",
       "17                                     64.562537  \n",
       "18                                     64.562537  \n",
       "19                                           NaN  \n",
       "20                                     71.710551  \n",
       "21                                   -111.603363  \n",
       "22                                    -86.887739  \n",
       "23                                     69.689377  \n",
       "24                                           NaN  \n",
       "25                                     64.562537  \n",
       "26                                   -137.774399  \n",
       "27                                   -114.248528  \n",
       "28                                    -73.496846  \n",
       "29                                    -82.454814  \n",
       "..                                           ...  \n",
       "792                                   -69.744626  \n",
       "793                                          NaN  \n",
       "794                                    -4.145546  \n",
       "795                                    64.562537  \n",
       "796                                  -112.818256  \n",
       "797                                  -150.695800  \n",
       "798                                  -127.310923  \n",
       "799                                   -82.454814  \n",
       "800                                    64.562537  \n",
       "801                                    16.847326  \n",
       "802                                   -82.454814  \n",
       "803                                   -92.915797  \n",
       "804                                    75.796797  \n",
       "805                                          NaN  \n",
       "806                                          NaN  \n",
       "807                                          NaN  \n",
       "808                                   -38.208687  \n",
       "809                                  -114.248528  \n",
       "810                                  -137.896881  \n",
       "811                                  -137.896881  \n",
       "812                                  -120.652510  \n",
       "813                                  -112.818256  \n",
       "814                                    61.449162  \n",
       "815                                   -99.729821  \n",
       "816                                   -79.321465  \n",
       "817                                          NaN  \n",
       "818                                  -112.818256  \n",
       "819                                  -120.652510  \n",
       "820                                  -137.896881  \n",
       "821                                    61.449162  \n",
       "\n",
       "[822 rows x 96 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# subset from the metagenomic data:\n",
    "metag_tmp = pd.read_csv(\"ENA_tables/PRJEB4352_metaG_wenv_PE.txt\", sep = '\\t')\n",
    "\n",
    "# repeat for metatranscriptomic data\n",
    "# metat_tmp = pd.read_csv(\"ENA_tables/PRJEB6609_metaT_wenv_PE.txt\", sep = '\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/vortexfs1/apps/anaconda-5.1/lib/python3.6/site-packages/ipykernel_launcher.py:12: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  if sys.path[0] == '':\n"
     ]
    }
   ],
   "source": [
    "# subset from the metagenomic data:\n",
    "metag_tmp = pd.read_csv(\"ENA_tables/PRJEB4352_metaG_wenv_PE.txt\", sep = '\\t')\n",
    "metag_tmp_subset_1 = metag_tmp[metag_tmp['OS region'] == \"[IO] Indian Ocean (MRGID:1904)\"].sample(n=10)\n",
    "metag_tmp_subset_2 = metag_tmp[metag_tmp['OS region'] == \"[MS] Mediterranean Sea (MRGID:1905)\"].sample(n=10)\n",
    "metag_cat_subset = pd.concat([metag_tmp_subset_1, metag_tmp_subset_2])\n",
    "metag_cat_subset.to_csv('ENA_tables/PRJEB4352_metaG_wenv_PE-TEST.txt', sep = '\\t')\n",
    "\n",
    "# repeat for metatranscriptomic data\n",
    "metat_tmp = pd.read_csv(\"ENA_tables/PRJEB6609_metaT_wenv_PE.txt\", sep = '\\t')\n",
    "metat_tmp_subset_1 = metat_tmp[metat_tmp['OS region'] == \"[IO] Indian Ocean (MRGID:1904)\"].sample(n=10)\n",
    "metat_tmp_subset_2 = metat_tmp[metat_tmp['OS region'] == \"[MS] Mediterranean Sea (MRGID:1905)\"].sample(n=10)\n",
    "metat_cat_subset = pd.concat([metat_tmp_subset_1, metag_tmp_subset_2])\n",
    "metat_cat_subset.to_csv('ENA_tables/PRJEB6609_metaT_wenv_PE-TEST.txt', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Excess below?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] File b'SampleList_ForAssembly.txt' does not exist: b'SampleList_ForAssembly.txt'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-44-198846e5b37b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0msara\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'SampleList_ForAssembly.txt'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'\\t'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex_col\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'Sub_region'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/.conda/envs/r_3.5.1/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36mparser_f\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, tupleize_cols, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)\u001b[0m\n\u001b[1;32m    700\u001b[0m                     skip_blank_lines=skip_blank_lines)\n\u001b[1;32m    701\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 702\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    703\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    704\u001b[0m     \u001b[0mparser_f\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.conda/envs/r_3.5.1/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    427\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    428\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 429\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    430\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    431\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.conda/envs/r_3.5.1/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m    893\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'has_index_names'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'has_index_names'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    894\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 895\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    896\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    897\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.conda/envs/r_3.5.1/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[0;34m(self, engine)\u001b[0m\n\u001b[1;32m   1120\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'c'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1121\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'c'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1122\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCParserWrapper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1123\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1124\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'python'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.conda/envs/r_3.5.1/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, src, **kwds)\u001b[0m\n\u001b[1;32m   1851\u001b[0m         \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'usecols'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0musecols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1852\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1853\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparsers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTextReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msrc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1854\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1855\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader.__cinit__\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._setup_parser_source\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] File b'SampleList_ForAssembly.txt' does not exist: b'SampleList_ForAssembly.txt'"
     ]
    }
   ],
   "source": [
    "sara = pd.read_csv('SampleList_ForAssembly.txt', sep = '\\t', index_col = 'Sub_region')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'sara' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-45-0faa28ce1673>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmetag_out\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDepth_sizefrac\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munique\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdifference\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msara\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDepth_sizefrac\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munique\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'sara' is not defined"
     ]
    }
   ],
   "source": [
    "set(metag_out.Depth_sizefrac.unique()).difference(set(sara.Depth_sizefrac.unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'0.8-5.00', '180-2000.00', '20-180.00', '5-20.00'}"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(metag.size_frac)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Depth_sizefrac</th>\n",
       "      <th>ERR_count</th>\n",
       "      <th>ERR_list</th>\n",
       "      <th>Assembly_group</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sub_region</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>DCM-0.8-5.00</td>\n",
       "      <td>13</td>\n",
       "      <td>ERR1726633, ERR1726897, ERR599323, ERR599324, ...</td>\n",
       "      <td>IO-all-DCM-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>DCM-180-2000.00</td>\n",
       "      <td>12</td>\n",
       "      <td>ERR1726526, ERR1726686, ERR1726696, ERR1726746...</td>\n",
       "      <td>IO-all-DCM-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>DCM-20-180.00</td>\n",
       "      <td>10</td>\n",
       "      <td>ERR1726554, ERR1726590, ERR1726671, ERR1726736...</td>\n",
       "      <td>IO-all-DCM-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>DCM-5-20.00</td>\n",
       "      <td>6</td>\n",
       "      <td>ERR1700904, ERR1700912, ERR1726665, ERR599205,...</td>\n",
       "      <td>IO-all-DCM-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>FSW-0.8-5.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR868368, ERR868484</td>\n",
       "      <td>IO-all-FSW-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>MES-0.8-5.00</td>\n",
       "      <td>8</td>\n",
       "      <td>ERR599212, ERR599225, ERR599268, ERR868353, ER...</td>\n",
       "      <td>IO-all-MES-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>SRF-0.8-5.00</td>\n",
       "      <td>20</td>\n",
       "      <td>ERR1726549, ERR1740328, ERR599210, ERR599266, ...</td>\n",
       "      <td>IO-all-SRF-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>SRF-180-2000.00</td>\n",
       "      <td>21</td>\n",
       "      <td>ERR1700889, ERR1700906, ERR1700907, ERR1700909...</td>\n",
       "      <td>IO-all-SRF-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>17</td>\n",
       "      <td>ERR1700914, ERR1726587, ERR1726599, ERR1726611...</td>\n",
       "      <td>IO-all-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>12</td>\n",
       "      <td>ERR1700890, ERR1700899, ERR1726575, ERR1726677...</td>\n",
       "      <td>IO-all-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IO-all</th>\n",
       "      <td>ZZZ-0.8-5.00</td>\n",
       "      <td>6</td>\n",
       "      <td>ERR1700892, ERR1726582, ERR1726591, ERR1726862...</td>\n",
       "      <td>IO-all-ZZZ-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MS-all</th>\n",
       "      <td>DCM-0.8-5.00</td>\n",
       "      <td>10</td>\n",
       "      <td>ERR538175, ERR538179, ERR538182, ERR868373, ER...</td>\n",
       "      <td>MS-all-DCM-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MS-all</th>\n",
       "      <td>DCM-180-2000.00</td>\n",
       "      <td>8</td>\n",
       "      <td>ERR1726557, ERR1726618, ERR1726672, ERR1726718...</td>\n",
       "      <td>MS-all-DCM-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MS-all</th>\n",
       "      <td>DCM-20-180.00</td>\n",
       "      <td>7</td>\n",
       "      <td>ERR1726743, ERR1726811, ERR1726836, ERR1726855...</td>\n",
       "      <td>MS-all-DCM-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MS-all</th>\n",
       "      <td>DCM-5-20.00</td>\n",
       "      <td>7</td>\n",
       "      <td>ERR1700894, ERR1700913, ERR1726585, ERR1726715...</td>\n",
       "      <td>MS-all-DCM-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MS-all</th>\n",
       "      <td>SRF-0.8-5.00</td>\n",
       "      <td>14</td>\n",
       "      <td>ERR1726597, ERR538173, ERR538186, ERR868356, E...</td>\n",
       "      <td>MS-all-SRF-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MS-all</th>\n",
       "      <td>SRF-180-2000.00</td>\n",
       "      <td>15</td>\n",
       "      <td>ERR1726574, ERR1726612, ERR1726662, ERR1726681...</td>\n",
       "      <td>MS-all-SRF-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MS-all</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>13</td>\n",
       "      <td>ERR1726546, ERR1726613, ERR1726642, ERR1726660...</td>\n",
       "      <td>MS-all-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MS-all</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>9</td>\n",
       "      <td>ERR1700895, ERR1700896, ERR1726540, ERR1726555...</td>\n",
       "      <td>MS-all-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>DCM-0.8-5.00</td>\n",
       "      <td>8</td>\n",
       "      <td>ERR1726596, ERR868367, ERR868423, ERR868435, E...</td>\n",
       "      <td>NAO-all-DCM-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>DCM-180-2000.00</td>\n",
       "      <td>4</td>\n",
       "      <td>ERR1726617, ERR1726783, ERR1726824, ERR1726957</td>\n",
       "      <td>NAO-all-DCM-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>DCM-20-180.00</td>\n",
       "      <td>4</td>\n",
       "      <td>ERR1726529, ERR1726852, ERR1726893, ERR1726911</td>\n",
       "      <td>NAO-all-DCM-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>DCM-5-20.00</td>\n",
       "      <td>5</td>\n",
       "      <td>ERR1700891, ERR1700900, ERR1700908, ERR1726551...</td>\n",
       "      <td>NAO-all-DCM-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>MES-0.8-5.00</td>\n",
       "      <td>9</td>\n",
       "      <td>ERR868349, ERR868380, ERR868434, ERR868449, ER...</td>\n",
       "      <td>NAO-all-MES-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>MIX-0.8-5.00</td>\n",
       "      <td>9</td>\n",
       "      <td>ERR1726629, ERR1726707, ERR1726761, ERR1726821...</td>\n",
       "      <td>NAO-all-MIX-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>MIX-180-2000.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726532, ERR1726941</td>\n",
       "      <td>NAO-all-MIX-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>MIX-20-180.00</td>\n",
       "      <td>3</td>\n",
       "      <td>ERR1726653, ERR1726803, ERR1726947</td>\n",
       "      <td>NAO-all-MIX-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>MIX-5-20.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726545, ERR1726756</td>\n",
       "      <td>NAO-all-MIX-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>SRF-0.8-5.00</td>\n",
       "      <td>20</td>\n",
       "      <td>ERR1726589, ERR1726676, ERR1726679, ERR1726693...</td>\n",
       "      <td>NAO-all-SRF-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAO-all</th>\n",
       "      <td>SRF-180-2000.00</td>\n",
       "      <td>13</td>\n",
       "      <td>ERR1726553, ERR1726638, ERR1726654, ERR1726712...</td>\n",
       "      <td>NAO-all-SRF-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SO-all</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>5</td>\n",
       "      <td>ERR1726781, ERR1726799, ERR1726901, ERR599264,...</td>\n",
       "      <td>SO-all-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SO-all</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>4</td>\n",
       "      <td>ERR1726939, ERR1726968, ERR599224, ERR599228</td>\n",
       "      <td>SO-all-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>DCM-0.8-5.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR868358, ERR868505</td>\n",
       "      <td>SPO-CHIL-DCM-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>DCM-180-2000.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726566, ERR1726627</td>\n",
       "      <td>SPO-CHIL-DCM-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>DCM-20-180.00</td>\n",
       "      <td>1</td>\n",
       "      <td>ERR1726830</td>\n",
       "      <td>SPO-CHIL-DCM-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>DCM-5-20.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1700901, ERR1726828</td>\n",
       "      <td>SPO-CHIL-DCM-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>MES-0.8-5.00</td>\n",
       "      <td>1</td>\n",
       "      <td>ERR868377</td>\n",
       "      <td>SPO-CHIL-MES-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>SRF-0.8-5.00</td>\n",
       "      <td>6</td>\n",
       "      <td>ERR1726645, ERR1726719, ERR1726749, ERR868413,...</td>\n",
       "      <td>SPO-CHIL-SRF-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>SRF-180-2000.00</td>\n",
       "      <td>4</td>\n",
       "      <td>ERR1726740, ERR1726790, ERR1726876, ERR1726895</td>\n",
       "      <td>SPO-CHIL-SRF-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>4</td>\n",
       "      <td>ERR1726524, ERR1726525, ERR1726850, ERR1726952</td>\n",
       "      <td>SPO-CHIL-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-CHIL</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>3</td>\n",
       "      <td>ERR1726632, ERR1726713, ERR1726878</td>\n",
       "      <td>SPO-CHIL-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>DCM-0.8-5.00</td>\n",
       "      <td>5</td>\n",
       "      <td>ERR1726842, ERR1726858, ERR868422, ERR868440, ...</td>\n",
       "      <td>SPO-PEOD-DCM-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>DCM-180-2000.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1700911, ERR1726962</td>\n",
       "      <td>SPO-PEOD-DCM-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>DCM-20-180.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726659, ERR1726727</td>\n",
       "      <td>SPO-PEOD-DCM-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>DCM-5-20.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726562, ERR1726742</td>\n",
       "      <td>SPO-PEOD-DCM-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>MES-0.8-5.00</td>\n",
       "      <td>1</td>\n",
       "      <td>ERR868362</td>\n",
       "      <td>SPO-PEOD-MES-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>SRF-0.8-5.00</td>\n",
       "      <td>4</td>\n",
       "      <td>ERR1726628, ERR1726851, ERR868357, ERR868462</td>\n",
       "      <td>SPO-PEOD-SRF-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>SRF-180-2000.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726705, ERR1726795</td>\n",
       "      <td>SPO-PEOD-SRF-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726577, ERR1726709</td>\n",
       "      <td>SPO-PEOD-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-PEOD</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>2</td>\n",
       "      <td>ERR1726594, ERR1726841</td>\n",
       "      <td>SPO-PEOD-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>DCM-0.8-5.00</td>\n",
       "      <td>19</td>\n",
       "      <td>ERR1726527, ERR1726579, ERR1726584, ERR1726609...</td>\n",
       "      <td>SPO-SPSG-DCM-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>DCM-180-2000.00</td>\n",
       "      <td>12</td>\n",
       "      <td>ERR1726550, ERR1726552, ERR1726604, ERR1726623...</td>\n",
       "      <td>SPO-SPSG-DCM-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>DCM-20-180.00</td>\n",
       "      <td>7</td>\n",
       "      <td>ERR1726603, ERR1726606, ERR1726655, ERR1726690...</td>\n",
       "      <td>SPO-SPSG-DCM-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>DCM-5-20.00</td>\n",
       "      <td>9</td>\n",
       "      <td>ERR1700897, ERR1700903, ERR1700905, ERR1726668...</td>\n",
       "      <td>SPO-SPSG-DCM-5-20.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>MES-0.8-5.00</td>\n",
       "      <td>7</td>\n",
       "      <td>ERR868365, ERR868379, ERR868417, ERR868443, ER...</td>\n",
       "      <td>SPO-SPSG-MES-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>MIX-0.8-5.00</td>\n",
       "      <td>7</td>\n",
       "      <td>ERR868359, ERR868360, ERR868371, ERR868426, ER...</td>\n",
       "      <td>SPO-SPSG-MIX-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>SRF-0.8-5.00</td>\n",
       "      <td>33</td>\n",
       "      <td>ERR1726556, ERR1726564, ERR1726567, ERR1726569...</td>\n",
       "      <td>SPO-SPSG-SRF-0.8-5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>SRF-180-2000.00</td>\n",
       "      <td>27</td>\n",
       "      <td>ERR1726541, ERR1726543, ERR1726561, ERR1726586...</td>\n",
       "      <td>SPO-SPSG-SRF-180-2000.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>SRF-20-180.00</td>\n",
       "      <td>25</td>\n",
       "      <td>ERR1726528, ERR1726544, ERR1726558, ERR1726588...</td>\n",
       "      <td>SPO-SPSG-SRF-20-180.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPO-SPSG</th>\n",
       "      <td>SRF-5-20.00</td>\n",
       "      <td>18</td>\n",
       "      <td>ERR1726522, ERR1726533, ERR1726535, ERR1726571...</td>\n",
       "      <td>SPO-SPSG-SRF-5-20.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>94 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Depth_sizefrac  ERR_count  \\\n",
       "Sub_region                               \n",
       "IO-all         DCM-0.8-5.00         13   \n",
       "IO-all      DCM-180-2000.00         12   \n",
       "IO-all        DCM-20-180.00         10   \n",
       "IO-all          DCM-5-20.00          6   \n",
       "IO-all         FSW-0.8-5.00          2   \n",
       "IO-all         MES-0.8-5.00          8   \n",
       "IO-all         SRF-0.8-5.00         20   \n",
       "IO-all      SRF-180-2000.00         21   \n",
       "IO-all        SRF-20-180.00         17   \n",
       "IO-all          SRF-5-20.00         12   \n",
       "IO-all         ZZZ-0.8-5.00          6   \n",
       "MS-all         DCM-0.8-5.00         10   \n",
       "MS-all      DCM-180-2000.00          8   \n",
       "MS-all        DCM-20-180.00          7   \n",
       "MS-all          DCM-5-20.00          7   \n",
       "MS-all         SRF-0.8-5.00         14   \n",
       "MS-all      SRF-180-2000.00         15   \n",
       "MS-all        SRF-20-180.00         13   \n",
       "MS-all          SRF-5-20.00          9   \n",
       "NAO-all        DCM-0.8-5.00          8   \n",
       "NAO-all     DCM-180-2000.00          4   \n",
       "NAO-all       DCM-20-180.00          4   \n",
       "NAO-all         DCM-5-20.00          5   \n",
       "NAO-all        MES-0.8-5.00          9   \n",
       "NAO-all        MIX-0.8-5.00          9   \n",
       "NAO-all     MIX-180-2000.00          2   \n",
       "NAO-all       MIX-20-180.00          3   \n",
       "NAO-all         MIX-5-20.00          2   \n",
       "NAO-all        SRF-0.8-5.00         20   \n",
       "NAO-all     SRF-180-2000.00         13   \n",
       "...                     ...        ...   \n",
       "SO-all        SRF-20-180.00          5   \n",
       "SO-all          SRF-5-20.00          4   \n",
       "SPO-CHIL       DCM-0.8-5.00          2   \n",
       "SPO-CHIL    DCM-180-2000.00          2   \n",
       "SPO-CHIL      DCM-20-180.00          1   \n",
       "SPO-CHIL        DCM-5-20.00          2   \n",
       "SPO-CHIL       MES-0.8-5.00          1   \n",
       "SPO-CHIL       SRF-0.8-5.00          6   \n",
       "SPO-CHIL    SRF-180-2000.00          4   \n",
       "SPO-CHIL      SRF-20-180.00          4   \n",
       "SPO-CHIL        SRF-5-20.00          3   \n",
       "SPO-PEOD       DCM-0.8-5.00          5   \n",
       "SPO-PEOD    DCM-180-2000.00          2   \n",
       "SPO-PEOD      DCM-20-180.00          2   \n",
       "SPO-PEOD        DCM-5-20.00          2   \n",
       "SPO-PEOD       MES-0.8-5.00          1   \n",
       "SPO-PEOD       SRF-0.8-5.00          4   \n",
       "SPO-PEOD    SRF-180-2000.00          2   \n",
       "SPO-PEOD      SRF-20-180.00          2   \n",
       "SPO-PEOD        SRF-5-20.00          2   \n",
       "SPO-SPSG       DCM-0.8-5.00         19   \n",
       "SPO-SPSG    DCM-180-2000.00         12   \n",
       "SPO-SPSG      DCM-20-180.00          7   \n",
       "SPO-SPSG        DCM-5-20.00          9   \n",
       "SPO-SPSG       MES-0.8-5.00          7   \n",
       "SPO-SPSG       MIX-0.8-5.00          7   \n",
       "SPO-SPSG       SRF-0.8-5.00         33   \n",
       "SPO-SPSG    SRF-180-2000.00         27   \n",
       "SPO-SPSG      SRF-20-180.00         25   \n",
       "SPO-SPSG        SRF-5-20.00         18   \n",
       "\n",
       "                                                     ERR_list  \\\n",
       "Sub_region                                                      \n",
       "IO-all      ERR1726633, ERR1726897, ERR599323, ERR599324, ...   \n",
       "IO-all      ERR1726526, ERR1726686, ERR1726696, ERR1726746...   \n",
       "IO-all      ERR1726554, ERR1726590, ERR1726671, ERR1726736...   \n",
       "IO-all      ERR1700904, ERR1700912, ERR1726665, ERR599205,...   \n",
       "IO-all                                   ERR868368, ERR868484   \n",
       "IO-all      ERR599212, ERR599225, ERR599268, ERR868353, ER...   \n",
       "IO-all      ERR1726549, ERR1740328, ERR599210, ERR599266, ...   \n",
       "IO-all      ERR1700889, ERR1700906, ERR1700907, ERR1700909...   \n",
       "IO-all      ERR1700914, ERR1726587, ERR1726599, ERR1726611...   \n",
       "IO-all      ERR1700890, ERR1700899, ERR1726575, ERR1726677...   \n",
       "IO-all      ERR1700892, ERR1726582, ERR1726591, ERR1726862...   \n",
       "MS-all      ERR538175, ERR538179, ERR538182, ERR868373, ER...   \n",
       "MS-all      ERR1726557, ERR1726618, ERR1726672, ERR1726718...   \n",
       "MS-all      ERR1726743, ERR1726811, ERR1726836, ERR1726855...   \n",
       "MS-all      ERR1700894, ERR1700913, ERR1726585, ERR1726715...   \n",
       "MS-all      ERR1726597, ERR538173, ERR538186, ERR868356, E...   \n",
       "MS-all      ERR1726574, ERR1726612, ERR1726662, ERR1726681...   \n",
       "MS-all      ERR1726546, ERR1726613, ERR1726642, ERR1726660...   \n",
       "MS-all      ERR1700895, ERR1700896, ERR1726540, ERR1726555...   \n",
       "NAO-all     ERR1726596, ERR868367, ERR868423, ERR868435, E...   \n",
       "NAO-all        ERR1726617, ERR1726783, ERR1726824, ERR1726957   \n",
       "NAO-all        ERR1726529, ERR1726852, ERR1726893, ERR1726911   \n",
       "NAO-all     ERR1700891, ERR1700900, ERR1700908, ERR1726551...   \n",
       "NAO-all     ERR868349, ERR868380, ERR868434, ERR868449, ER...   \n",
       "NAO-all     ERR1726629, ERR1726707, ERR1726761, ERR1726821...   \n",
       "NAO-all                                ERR1726532, ERR1726941   \n",
       "NAO-all                    ERR1726653, ERR1726803, ERR1726947   \n",
       "NAO-all                                ERR1726545, ERR1726756   \n",
       "NAO-all     ERR1726589, ERR1726676, ERR1726679, ERR1726693...   \n",
       "NAO-all     ERR1726553, ERR1726638, ERR1726654, ERR1726712...   \n",
       "...                                                       ...   \n",
       "SO-all      ERR1726781, ERR1726799, ERR1726901, ERR599264,...   \n",
       "SO-all           ERR1726939, ERR1726968, ERR599224, ERR599228   \n",
       "SPO-CHIL                                 ERR868358, ERR868505   \n",
       "SPO-CHIL                               ERR1726566, ERR1726627   \n",
       "SPO-CHIL                                           ERR1726830   \n",
       "SPO-CHIL                               ERR1700901, ERR1726828   \n",
       "SPO-CHIL                                            ERR868377   \n",
       "SPO-CHIL    ERR1726645, ERR1726719, ERR1726749, ERR868413,...   \n",
       "SPO-CHIL       ERR1726740, ERR1726790, ERR1726876, ERR1726895   \n",
       "SPO-CHIL       ERR1726524, ERR1726525, ERR1726850, ERR1726952   \n",
       "SPO-CHIL                   ERR1726632, ERR1726713, ERR1726878   \n",
       "SPO-PEOD    ERR1726842, ERR1726858, ERR868422, ERR868440, ...   \n",
       "SPO-PEOD                               ERR1700911, ERR1726962   \n",
       "SPO-PEOD                               ERR1726659, ERR1726727   \n",
       "SPO-PEOD                               ERR1726562, ERR1726742   \n",
       "SPO-PEOD                                            ERR868362   \n",
       "SPO-PEOD         ERR1726628, ERR1726851, ERR868357, ERR868462   \n",
       "SPO-PEOD                               ERR1726705, ERR1726795   \n",
       "SPO-PEOD                               ERR1726577, ERR1726709   \n",
       "SPO-PEOD                               ERR1726594, ERR1726841   \n",
       "SPO-SPSG    ERR1726527, ERR1726579, ERR1726584, ERR1726609...   \n",
       "SPO-SPSG    ERR1726550, ERR1726552, ERR1726604, ERR1726623...   \n",
       "SPO-SPSG    ERR1726603, ERR1726606, ERR1726655, ERR1726690...   \n",
       "SPO-SPSG    ERR1700897, ERR1700903, ERR1700905, ERR1726668...   \n",
       "SPO-SPSG    ERR868365, ERR868379, ERR868417, ERR868443, ER...   \n",
       "SPO-SPSG    ERR868359, ERR868360, ERR868371, ERR868426, ER...   \n",
       "SPO-SPSG    ERR1726556, ERR1726564, ERR1726567, ERR1726569...   \n",
       "SPO-SPSG    ERR1726541, ERR1726543, ERR1726561, ERR1726586...   \n",
       "SPO-SPSG    ERR1726528, ERR1726544, ERR1726558, ERR1726588...   \n",
       "SPO-SPSG    ERR1726522, ERR1726533, ERR1726535, ERR1726571...   \n",
       "\n",
       "                      Assembly_group  \n",
       "Sub_region                            \n",
       "IO-all           IO-all-DCM-0.8-5.00  \n",
       "IO-all        IO-all-DCM-180-2000.00  \n",
       "IO-all          IO-all-DCM-20-180.00  \n",
       "IO-all            IO-all-DCM-5-20.00  \n",
       "IO-all           IO-all-FSW-0.8-5.00  \n",
       "IO-all           IO-all-MES-0.8-5.00  \n",
       "IO-all           IO-all-SRF-0.8-5.00  \n",
       "IO-all        IO-all-SRF-180-2000.00  \n",
       "IO-all          IO-all-SRF-20-180.00  \n",
       "IO-all            IO-all-SRF-5-20.00  \n",
       "IO-all           IO-all-ZZZ-0.8-5.00  \n",
       "MS-all           MS-all-DCM-0.8-5.00  \n",
       "MS-all        MS-all-DCM-180-2000.00  \n",
       "MS-all          MS-all-DCM-20-180.00  \n",
       "MS-all            MS-all-DCM-5-20.00  \n",
       "MS-all           MS-all-SRF-0.8-5.00  \n",
       "MS-all        MS-all-SRF-180-2000.00  \n",
       "MS-all          MS-all-SRF-20-180.00  \n",
       "MS-all            MS-all-SRF-5-20.00  \n",
       "NAO-all         NAO-all-DCM-0.8-5.00  \n",
       "NAO-all      NAO-all-DCM-180-2000.00  \n",
       "NAO-all        NAO-all-DCM-20-180.00  \n",
       "NAO-all          NAO-all-DCM-5-20.00  \n",
       "NAO-all         NAO-all-MES-0.8-5.00  \n",
       "NAO-all         NAO-all-MIX-0.8-5.00  \n",
       "NAO-all      NAO-all-MIX-180-2000.00  \n",
       "NAO-all        NAO-all-MIX-20-180.00  \n",
       "NAO-all          NAO-all-MIX-5-20.00  \n",
       "NAO-all         NAO-all-SRF-0.8-5.00  \n",
       "NAO-all      NAO-all-SRF-180-2000.00  \n",
       "...                              ...  \n",
       "SO-all          SO-all-SRF-20-180.00  \n",
       "SO-all            SO-all-SRF-5-20.00  \n",
       "SPO-CHIL       SPO-CHIL-DCM-0.8-5.00  \n",
       "SPO-CHIL    SPO-CHIL-DCM-180-2000.00  \n",
       "SPO-CHIL      SPO-CHIL-DCM-20-180.00  \n",
       "SPO-CHIL        SPO-CHIL-DCM-5-20.00  \n",
       "SPO-CHIL       SPO-CHIL-MES-0.8-5.00  \n",
       "SPO-CHIL       SPO-CHIL-SRF-0.8-5.00  \n",
       "SPO-CHIL    SPO-CHIL-SRF-180-2000.00  \n",
       "SPO-CHIL      SPO-CHIL-SRF-20-180.00  \n",
       "SPO-CHIL        SPO-CHIL-SRF-5-20.00  \n",
       "SPO-PEOD       SPO-PEOD-DCM-0.8-5.00  \n",
       "SPO-PEOD    SPO-PEOD-DCM-180-2000.00  \n",
       "SPO-PEOD      SPO-PEOD-DCM-20-180.00  \n",
       "SPO-PEOD        SPO-PEOD-DCM-5-20.00  \n",
       "SPO-PEOD       SPO-PEOD-MES-0.8-5.00  \n",
       "SPO-PEOD       SPO-PEOD-SRF-0.8-5.00  \n",
       "SPO-PEOD    SPO-PEOD-SRF-180-2000.00  \n",
       "SPO-PEOD      SPO-PEOD-SRF-20-180.00  \n",
       "SPO-PEOD        SPO-PEOD-SRF-5-20.00  \n",
       "SPO-SPSG       SPO-SPSG-DCM-0.8-5.00  \n",
       "SPO-SPSG    SPO-SPSG-DCM-180-2000.00  \n",
       "SPO-SPSG      SPO-SPSG-DCM-20-180.00  \n",
       "SPO-SPSG        SPO-SPSG-DCM-5-20.00  \n",
       "SPO-SPSG       SPO-SPSG-MES-0.8-5.00  \n",
       "SPO-SPSG       SPO-SPSG-MIX-0.8-5.00  \n",
       "SPO-SPSG       SPO-SPSG-SRF-0.8-5.00  \n",
       "SPO-SPSG    SPO-SPSG-SRF-180-2000.00  \n",
       "SPO-SPSG      SPO-SPSG-SRF-20-180.00  \n",
       "SPO-SPSG        SPO-SPSG-SRF-5-20.00  \n",
       "\n",
       "[94 rows x 4 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sara"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "94"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
